{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build and annotate INs\n",
    "Shows how to use the pipeline that first builds Insulated Neighborhoods using a Louvian Graph based partitioning algorithm and then annotates all Cis-regulatory elements using a Neural Network. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: buildAnnotatteINs.py [-h] -i INPUT_DIRECTORY -o OUTPUT_DIRECTORY\r\n",
      "                            [-t INSULATED_NEIGH] -c DNA_INT_CELL_TYPE -q\r\n",
      "                            EQTL_TISSUE [-g TOP_GENE_FPKM] -n NAME_OUTPUT\r\n",
      "                            [-m MARKER_REGION]\r\n",
      "\r\n",
      "Pipeline to output graphical genomic model from 2D data\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -i INPUT_DIRECTORY, --input_directory INPUT_DIRECTORY\r\n",
      "                        Where all the input data lives\r\n",
      "  -o OUTPUT_DIRECTORY, --output_directory OUTPUT_DIRECTORY\r\n",
      "                        Output directory of graphs\r\n",
      "  -t INSULATED_NEIGH, --insulated_neigh INSULATED_NEIGH\r\n",
      "                        Comma separated list of Insulated Neighborhoods to\r\n",
      "                        build\r\n",
      "  -c DNA_INT_CELL_TYPE, --dna_int_cell_type DNA_INT_CELL_TYPE\r\n",
      "                        Cell type to use in building Insulated Neighborhoods\r\n",
      "  -q EQTL_TISSUE, --eQTL_tissue EQTL_TISSUE\r\n",
      "                        What tissue from GTeX to do eQTL analysis w/.\r\n",
      "                        Available tissues:\r\n",
      "                        Adipose_Subcutaneous,Adipose_Visceral_Omentum,\r\n",
      "                        Adrenal_Gland, Artery_Aorta, Artery_Coronary,\r\n",
      "                        Artery_Tibial, Brain_Anterior_cingulate_cortex_BA24,\r\n",
      "                        Brain_Caudate_basal_ganglia,\r\n",
      "                        Brain_Cerebellar_Hemisphere, Brain_Cerebellum,\r\n",
      "                        Brain_Cortex, Brain_Frontal_Cortex_BA9,\r\n",
      "                        Brain_Hippocampus, Brain_Hypothalamus,\r\n",
      "                        Brain_Nucleus_accumbens_basal_ganglia,\r\n",
      "                        Brain_Putamen_basal_ganglia,Breast_Mammary_Tissue,\r\n",
      "                        Cells_EBV-transformed_lymphocytes,\r\n",
      "                        Cells_Transformed_fibroblasts, Colon_Sigmoid,\r\n",
      "                        Colon_Transverse, Esophagus_Gastroesophageal_Junction,\r\n",
      "                        Esophagus_Mucosa, Esophagus_Muscularis,\r\n",
      "                        Heart_Atrial_Appendage, Heart_Left_Ventricle, Liver,\r\n",
      "                        Lung, Muscle_Skeletal, Nerve_Tibial, Ovary, Pancreas,\r\n",
      "                        Pituitary, Prostate, Skin_Not_Sun_Exposed_Suprapubic,\r\n",
      "                        Skin_Sun_Exposed_Lower_leg,\r\n",
      "                        Small_Intestine_Terminal_Ileum, Spleen, Stomach,\r\n",
      "                        Testis, Thyroid, Uterus, Vagina, Whole_Blood\r\n",
      "  -g TOP_GENE_FPKM, --top_gene_fpkm TOP_GENE_FPKM\r\n",
      "                        Instead of defining Insulated Neighborhoods take genes\r\n",
      "                        expressed above FPKM cutoff\r\n",
      "  -n NAME_OUTPUT, --name_output NAME_OUTPUT\r\n",
      "                        Output prefix for all output files\r\n",
      "  -m MARKER_REGION, --marker_region MARKER_REGION\r\n",
      "                        Build all Insulated Neighborhoods of genes which fall\r\n",
      "                        within this chromosomal region (ex:\r\n",
      "                        chr1:54996039-55996039)\r\n"
     ]
    }
   ],
   "source": [
    "!python /root/Euplotid/pipelines/buildAnnotatteINs.py --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Error: No results matching query were found.\r\n"
     ]
    }
   ],
   "source": [
    "!luarocks install trepl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Target Genes: 6\n",
      "Target Genes to build Insulated Neighborhoods: RPGRIP1L,JB149426,FTO,Y_RNA,IRX3,CHD9\n",
      "Checking resolution parameter against CTCF-CTCF loop\n",
      "Annotating 4 Neighborhoods\n",
      "Annotating the following Insulated Neighborhoods: CHD9|RPGRIP1L,FTO|CHD9,CHD9,CHD9,Y_RNA,CHD9|IRX3,FTO,JB149426,FTO\n",
      "Annotating Insulated Neighborhood: IRX3,FTO,JB149426,FTO\n",
      "Annotating Open Regions\n",
      "Annotating Insulated Neighborhood: CHD9\n",
      "Annotating Open Regions\n",
      "Annotating Insulated Neighborhood: CHD9,CHD9,CHD9,Y_RNA,CHD9\n",
      "Annotating Open Regions\n",
      "Annotating Insulated Neighborhood: RPGRIP1L,FTO\n",
      "Annotating Open Regions\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-4...done.\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-1...done.\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-5...done.\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-2...done.\n",
      "Fetching 12 variant(s) . . .\n",
      "Fetching 17 variant(s) . . .\n",
      "Fetching 52 variant(s) . . .\n",
      "Fetching 18 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 29 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 34 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 49 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr16:53979176-53980851\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr16:53241256-53242869\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr16:53254174-53255315\n",
      "No results to return\n",
      "Fetching 54 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 77 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 23 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 83 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 608 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 58 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 88 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 2393 variant(s) . . .\n",
      "Basset SNP prediction of: chr16:53979176-53980851\n",
      "/root/Basset/src/basset_sad.py -f /input_dir/hg19.fa -l 600 -o /output_dir//primed/example//chr16_53979176_53980851_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/example//chr16_53979176_53980851_basset/open_variants.vcf\n",
      "No results to return\n",
      "Fetching 25 variant(s) . . .\n",
      "WARNING: Skipping chr16:g.53932919_53932920del - neither allele matches reference genome: TAA vs AAA\n",
      "WARNING: Skipping chr16:g.53932930_53932932del - neither allele matches reference genome: CTCT vs TCTT\n",
      "WARNING: Skipping chr16:g.53933428del - neither allele matches reference genome: TC vs CC\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/example//chr16_53979176_53980851_basset/model_in.h5 /output_dir//primed/example//chr16_53979176_53980851_basset/model_out.txt\n",
      "/opt/conda/bin/lua: /opt/conda/share/lua/5.2/luarocks/loader.lua:117: error loading module 'treplutils' from file '/opt/conda/bin/lua':\n",
      "\t/opt/conda/bin/lua:1: unexpected symbol near char(127)\n",
      "stack traceback:\n",
      "\t[C]: in function 'a_loader'\n",
      "\t/opt/conda/share/lua/5.2/luarocks/loader.lua:117: in function </opt/conda/share/lua/5.2/luarocks/loader.lua:114>\n",
      "\t(...tail calls...)\n",
      "\t[C]: in function 'require'\n",
      "\t/opt/conda/share/lua/5.2/trepl/init.lua:40: in main chunk\n",
      "\t[C]: in function 'require'\n",
      "\t/opt/conda/lib/luarocks/rocks/trepl/scm-1/bin/th:104: in main chunk\n",
      "\t[C]: in ?\n",
      "Traceback (most recent call last):\n",
      "  File \"/root/Basset/src/basset_sad.py\", line 222, in <module>\n",
      "    main()\n",
      "  File \"/root/Basset/src/basset_sad.py\", line 92, in main\n",
      "    for line in open(options.model_hdf5_file):\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/output_dir//primed/example//chr16_53979176_53980851_basset/model_out.txt'\n",
      "multiprocessing.pool.RemoteTraceback: \n",
      "\"\"\"\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/joblib/_parallel_backends.py\", line 350, in __call__\n",
      "    return self.func(*args, **kwargs)\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/joblib/parallel.py\", line 131, in __call__\n",
      "    return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/joblib/parallel.py\", line 131, in <listcomp>\n",
      "    return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "  File \"/root/Euplotid/src/buildAnnotateINs.py\", line 781, in build_IN_levels\n",
      "    comm_var_out, comm_nodes_out, comm_edges_out, comm_edges_out_bed, ucsc_session_in,out_dir)\n",
      "  File \"/root/Euplotid/src/buildAnnotateINs.py\", line 673, in annotate_IN\n",
      "    source_community_graph = add_variants_predict(source_community_graph, homo_gen, chain_file, TF_RBP_ids, picked_tissue, target_node_name,out_dir)\n",
      "  File \"/root/Euplotid/src/buildAnnotateINs.py\", line 302, in add_variants_predict\n",
      "    G = add_basset_sad_sat(G, homo_gen, target_node_name,out_dir)\n",
      "  File \"/root/Euplotid/src/buildAnnotateINs.py\", line 406, in add_basset_sad_sat\n",
      "    sad_table = pd.read_table(basset_out_name + \"/sad_table.txt\", delim_whitespace=True, header = 0)\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py\", line 655, in parser_f\n",
      "    return _read(filepath_or_buffer, kwds)\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py\", line 405, in _read\n",
      "    parser = TextFileReader(filepath_or_buffer, **kwds)\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py\", line 762, in __init__\n",
      "    self._make_engine(self.engine)\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py\", line 966, in _make_engine\n",
      "    self._engine = CParserWrapper(self.f, **self.options)\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py\", line 1582, in __init__\n",
      "    self._reader = parsers.TextReader(src, **kwds)\n",
      "  File \"pandas/_libs/parsers.pyx\", line 394, in pandas._libs.parsers.TextReader.__cinit__ (pandas/_libs/parsers.c:4209)\n",
      "  File \"pandas/_libs/parsers.pyx\", line 710, in pandas._libs.parsers.TextReader._setup_parser_source (pandas/_libs/parsers.c:8873)\n",
      "FileNotFoundError: File b'/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt' does not exist\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.5/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/joblib/_parallel_backends.py\", line 359, in __call__\n",
      "    raise TransportableException(text, e_type)\n",
      "joblib.my_exceptions.TransportableException: TransportableException\n",
      "___________________________________________________________________________\n",
      "FileNotFoundError                                  Wed May 31 03:20:16 2017\n",
      "PID: 3714                               Python 3.5.3: /opt/conda/bin/python\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/joblib/parallel.py in __call__(self=<joblib.parallel.BatchedCalls object>)\n",
      "    126     def __init__(self, iterator_slice):\n",
      "    127         self.items = list(iterator_slice)\n",
      "    128         self._size = len(self.items)\n",
      "    129 \n",
      "    130     def __call__(self):\n",
      "--> 131         return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "        self.items = [(<function build_IN_levels>, ('chr16:53979176-53980851', [{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1475, 'chr10:100185094-100186268': 1475, 'chr10:100205605-100207134': 1475, 'chr10:100226542-100228626': 1475, 'chr10:100423690-100425020': 1475, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], <networkx.classes.graph.Graph object>, '/input_dir/hg19.fa', '/input_dir/hg19ToHg38.over.chain.gz', '/input_dir/hg38ToHg19.over.chain.gz', '/output_dir//primed/example/', 'example', '/output_dir//primed/example/example_global_graph.json', '/input_dir/TF_RBP_human.ids', 'open_peaks_nodes.bed', 'Adipose_Visceral_Omentum', {'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, '/input_dir/hES_hsa_400kb_xyz.bed', 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed'), {})]\n",
      "    132 \n",
      "    133     def __len__(self):\n",
      "    134         return self._size\n",
      "    135 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/joblib/parallel.py in <listcomp>(.0=<list_iterator object>)\n",
      "    126     def __init__(self, iterator_slice):\n",
      "    127         self.items = list(iterator_slice)\n",
      "    128         self._size = len(self.items)\n",
      "    129 \n",
      "    130     def __call__(self):\n",
      "--> 131         return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "        func = <function build_IN_levels>\n",
      "        args = ('chr16:53979176-53980851', [{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1475, 'chr10:100185094-100186268': 1475, 'chr10:100205605-100207134': 1475, 'chr10:100226542-100228626': 1475, 'chr10:100423690-100425020': 1475, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], <networkx.classes.graph.Graph object>, '/input_dir/hg19.fa', '/input_dir/hg19ToHg38.over.chain.gz', '/input_dir/hg38ToHg19.over.chain.gz', '/output_dir//primed/example/', 'example', '/output_dir//primed/example/example_global_graph.json', '/input_dir/TF_RBP_human.ids', 'open_peaks_nodes.bed', 'Adipose_Visceral_Omentum', {'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, '/input_dir/hES_hsa_400kb_xyz.bed', 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed')\n",
      "        kwargs = {}\n",
      "    132 \n",
      "    133     def __len__(self):\n",
      "    134         return self._size\n",
      "    135 \n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in build_IN_levels(target_node_name='chr16:53979176-53980851', dendogram_com=[{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1528, 'chr10:100185094-100186268': 1528, 'chr10:100205605-100207134': 7231, 'chr10:100226542-100228626': 7231, 'chr10:100423690-100425020': 1528, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], dna_int_graph=<networkx.classes.graph.Graph object>, genome_fa='/input_dir/hg19.fa', chain_file='/input_dir/hg19ToHg38.over.chain.gz', chain_file2='/input_dir/hg38ToHg19.over.chain.gz', out_dir='/output_dir//primed/example/', run_name='example', json_draft_name='/output_dir//primed/example/example_global_graph.json', TF_RBP_ids='/input_dir/TF_RBP_human.ids', open_peaks_file='open_peaks_nodes.bed', picked_tissue='Adipose_Visceral_Omentum', ctcf2bounds={'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, hsa_3D_loc='/input_dir/hES_hsa_400kb_xyz.bed', ucsc_session='http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed')\n",
      "    776         add_snp = True\n",
      "    777     else:\n",
      "    778         add_snp = False\n",
      "    779     source_community_graph = annotate_IN(target_node_name, dendogram_com, level, dna_int_graph, add_open, add_snp,\n",
      "    780                                                   homo_gen, chain_file, chain_file2, TF_RBP_ids, open_peaks_file, picked_tissue, ctcf2bounds, comm_open_out,\n",
      "--> 781                                                   comm_var_out, comm_nodes_out, comm_edges_out, comm_edges_out_bed, ucsc_session_in,out_dir)\n",
      "        comm_var_out = <_io.TextIOWrapper name='/output_dir//primed/exa...y_variants_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        comm_nodes_out = <_io.TextIOWrapper name='/output_dir//primed/exa...nity_nodes_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        comm_edges_out = <_io.TextIOWrapper name='/output_dir//primed/exa...ty_edges_lvl_1.washu' mode='a+' encoding='UTF-8'>\n",
      "        comm_edges_out_bed = <_io.TextIOWrapper name='/output_dir//primed/exa...nity_edges_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        ucsc_session_in = 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...ssionName=primed&position=chr16:54668601-53806737'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    782     #level 1 \n",
      "    783     if len(dendogram_com) > 1:\n",
      "    784         level = 1\n",
      "    785         add_open = False\n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in annotate_IN(target_node_name='chr16:53979176-53980851', dendogram_com=[{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1528, 'chr10:100185094-100186268': 1528, 'chr10:100205605-100207134': 7231, 'chr10:100226542-100228626': 7231, 'chr10:100423690-100425020': 1528, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], level=0, dna_int_graph=<networkx.classes.graph.Graph object>, add_open=True, add_snp=True, homo_gen=<pysam.libcfaidx.FastaFile object>, chain_file='/input_dir/hg19ToHg38.over.chain.gz', chain_file2='/input_dir/hg38ToHg19.over.chain.gz', TF_RBP_ids='/input_dir/TF_RBP_human.ids', open_peaks_file='open_peaks_nodes.bed', picked_tissue='Adipose_Visceral_Omentum', ctcf2bounds={'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, comm_open_out=<_io.TextIOWrapper name='/output_dir//primed/exa...en_regions_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_var_out=<_io.TextIOWrapper name='/output_dir//primed/exa...y_variants_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_nodes_out=<_io.TextIOWrapper name='/output_dir//primed/exa...nity_nodes_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_edges_out=<_io.TextIOWrapper name='/output_dir//primed/exa...ty_edges_lvl_1.washu' mode='a+' encoding='UTF-8'>, comm_edges_out_bed=<_io.TextIOWrapper name='/output_dir//primed/exa...nity_edges_lvl_1.bed' mode='a+' encoding='UTF-8'>, ucsc_session_in='http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...ssionName=primed&position=chr16:54668601-53806737', out_dir='/output_dir//primed/example/')\n",
      "    668         print(\"Annotating Open Regions\")\n",
      "    669         source_community_graph = add_openRegions_predict(source_community_graph, homo_gen, chain_file, chain_file2, TF_RBP_ids, open_peaks_file, target_node_name, level, dict())\n",
      "    670         sys.stdout.flush()\n",
      "    671         if add_snp:\n",
      "    672             print(\"Annotating SNPs/CNVs\")\n",
      "--> 673             source_community_graph = add_variants_predict(source_community_graph, homo_gen, chain_file, TF_RBP_ids, picked_tissue, target_node_name,out_dir)\n",
      "        source_community_graph = <networkx.classes.graph.Graph object>\n",
      "        homo_gen = <pysam.libcfaidx.FastaFile object>\n",
      "        chain_file = '/input_dir/hg19ToHg38.over.chain.gz'\n",
      "        TF_RBP_ids = '/input_dir/TF_RBP_human.ids'\n",
      "        picked_tissue = 'Adipose_Visceral_Omentum'\n",
      "        target_node_name = 'chr16:53979176-53980851'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    674             sys.stdout.flush()\n",
      "    675     \n",
      "    676     #Color Insulated Neighborhood according to node attributes, make mid distance to target node\n",
      "    677 \n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in add_variants_predict(G=<networkx.classes.graph.Graph object>, homo_gen=<pysam.libcfaidx.FastaFile object>, chain_file='/input_dir/hg19ToHg38.over.chain.gz', TF_RBP_ids='/input_dir/TF_RBP_human.ids', tissue_type='Adipose_Visceral_Omentum', target_node_name='chr16:53979176-53980851', out_dir='/output_dir//primed/example/')\n",
      "    297     #predict effect of all variation added to graph using DeepBind and Basset\n",
      "    298     if add_any_SNPs:\n",
      "    299         print(\"Deepbind SNP prediction of: \" + target_node_name)\n",
      "    300         G = deepbind_predict_SNPs(G, homo_gen, TF_RBP_ids,target_node_name)\n",
      "    301         print(\"Basset SNP prediction of: \" + target_node_name)\n",
      "--> 302         G = add_basset_sad_sat(G, homo_gen, target_node_name,out_dir)\n",
      "        G = <networkx.classes.graph.Graph object>\n",
      "        homo_gen = <pysam.libcfaidx.FastaFile object>\n",
      "        target_node_name = 'chr16:53979176-53980851'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    303     try:\n",
      "    304         os.remove(basset_name_out + \"open_variants.vcf\")\n",
      "    305     except OSError:\n",
      "    306         pass\n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in add_basset_sad_sat(G=<networkx.classes.graph.Graph object>, homo_gen=<pysam.libcfaidx.FastaFile object>, target_node_name='chr16:53979176-53980851', out_dir='/output_dir//primed/example/')\n",
      "    401             count += 1\n",
      "    402     cmd = (\"/root/Basset/src/basset_sad.py -f \" + (homo_gen.filename).decode(\"utf-8\") + \" -l 600 -o \" + basset_out_name + \" -t \" + targets_file + \" \" +  model_file + \" \" + basset_out_name + \"/open_variants.vcf\")\n",
      "    403     print(cmd)\n",
      "    404     os.system(cmd)\n",
      "    405     #Pick SNP maximizing for sum of Delta SAD across all cell types in trained model\n",
      "--> 406     sad_table = pd.read_table(basset_out_name + \"/sad_table.txt\", delim_whitespace=True, header = 0)\n",
      "        sad_table = undefined\n",
      "        basset_out_name = '/output_dir//primed/example//chr16_53979176_53980851_basset'\n",
      "    407     sad_table_dense = sad_table.pivot_table(index='rsid', columns='target', values='pred')\n",
      "    408     serial_sad_table = pickle.dumps(sad_table_dense, protocol=0)\n",
      "    409     G.graph[\"sad_table\"] = serial_sad_table\n",
      "    410     sad_table[\"sad_abs\"] = abs(sad_table[\"pred\"])\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in parser_f(filepath_or_buffer='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', sep='\\t', delimiter='\\t', header=0, names=None, index_col=None, usecols=None, squeeze=False, prefix=None, mangle_dupe_cols=True, dtype=None, engine='c', converters=None, true_values=None, false_values=None, skipinitialspace=False, skiprows=None, nrows=None, na_values=None, keep_default_na=True, na_filter=True, verbose=False, skip_blank_lines=True, parse_dates=False, infer_datetime_format=False, keep_date_col=False, date_parser=None, dayfirst=False, iterator=False, chunksize=None, compression='infer', thousands=None, decimal=b'.', lineterminator=None, quotechar='\"', quoting=0, escapechar=None, comment=None, encoding=None, dialect=None, tupleize_cols=False, error_bad_lines=True, warn_bad_lines=True, skipfooter=0, skip_footer=0, doublequote=True, delim_whitespace=True, as_recarray=False, compact_ints=False, use_unsigned=False, low_memory=True, buffer_lines=None, memory_map=False, float_precision=None)\n",
      "    650                     mangle_dupe_cols=mangle_dupe_cols,\n",
      "    651                     tupleize_cols=tupleize_cols,\n",
      "    652                     infer_datetime_format=infer_datetime_format,\n",
      "    653                     skip_blank_lines=skip_blank_lines)\n",
      "    654 \n",
      "--> 655         return _read(filepath_or_buffer, kwds)\n",
      "        filepath_or_buffer = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...}\n",
      "    656 \n",
      "    657     parser_f.__name__ = name\n",
      "    658 \n",
      "    659     return parser_f\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in _read(filepath_or_buffer='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', kwds={'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...})\n",
      "    400     iterator = kwds.get('iterator', False)\n",
      "    401     chunksize = _validate_integer('chunksize', kwds.get('chunksize', None), 1)\n",
      "    402     nrows = _validate_integer('nrows', kwds.get('nrows', None))\n",
      "    403 \n",
      "    404     # Create the parser.\n",
      "--> 405     parser = TextFileReader(filepath_or_buffer, **kwds)\n",
      "        parser = undefined\n",
      "        filepath_or_buffer = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...}\n",
      "    406 \n",
      "    407     if chunksize or iterator:\n",
      "    408         return parser\n",
      "    409 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in __init__(self=<pandas.io.parsers.TextFileReader object>, f='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', engine='c', **kwds={'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...})\n",
      "    757         # might mutate self.engine\n",
      "    758         self.options, self.engine = self._clean_options(options, engine)\n",
      "    759         if 'has_index_names' in kwds:\n",
      "    760             self.options['has_index_names'] = kwds['has_index_names']\n",
      "    761 \n",
      "--> 762         self._make_engine(self.engine)\n",
      "        self._make_engine = <bound method TextFileReader._make_engine of <pandas.io.parsers.TextFileReader object>>\n",
      "        self.engine = 'c'\n",
      "    763 \n",
      "    764     def close(self):\n",
      "    765         self._engine.close()\n",
      "    766 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in _make_engine(self=<pandas.io.parsers.TextFileReader object>, engine='c')\n",
      "    961             self.close()\n",
      "    962             raise\n",
      "    963 \n",
      "    964     def _make_engine(self, engine='c'):\n",
      "    965         if engine == 'c':\n",
      "--> 966             self._engine = CParserWrapper(self.f, **self.options)\n",
      "        self._engine = None\n",
      "        self.f = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        self.options = {'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', 'delim_whitespace': True, ...}\n",
      "    967         else:\n",
      "    968             if engine == 'python':\n",
      "    969                 klass = PythonParser\n",
      "    970             elif engine == 'python-fwf':\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in __init__(self=<pandas.io.parsers.CParserWrapper object>, src='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', **kwds={'allow_leading_cols': True, 'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'decimal': b'.', 'delim_whitespace': True, 'delimiter': '\\t', ...})\n",
      "   1577             kwds['encoding'] = 'utf-8'\n",
      "   1578 \n",
      "   1579         # #2442\n",
      "   1580         kwds['allow_leading_cols'] = self.index_col is not False\n",
      "   1581 \n",
      "-> 1582         self._reader = parsers.TextReader(src, **kwds)\n",
      "        self._reader = undefined\n",
      "        src = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'allow_leading_cols': True, 'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'decimal': b'.', 'delim_whitespace': True, 'delimiter': '\\t', ...}\n",
      "   1583 \n",
      "   1584         # XXX\n",
      "   1585         self.usecols, self.usecols_dtype = _validate_usecols_arg(\n",
      "   1586             self._reader.usecols)\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/_libs/parsers.cpython-35m-x86_64-linux-gnu.so in pandas._libs.parsers.TextReader.__cinit__ (pandas/_libs/parsers.c:4209)()\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/_libs/parsers.cpython-35m-x86_64-linux-gnu.so in pandas._libs.parsers.TextReader._setup_parser_source (pandas/_libs/parsers.c:8873)()\n",
      "\n",
      "FileNotFoundError: File b'/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt' does not exist\n",
      "___________________________________________________________________________\n",
      "\"\"\"\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/joblib/parallel.py\", line 699, in retrieve\n",
      "    self._output.extend(job.get(timeout=self.timeout))\n",
      "  File \"/opt/conda/lib/python3.5/multiprocessing/pool.py\", line 608, in get\n",
      "    raise self._value\n",
      "joblib.my_exceptions.TransportableException: TransportableException\n",
      "___________________________________________________________________________\n",
      "FileNotFoundError                                  Wed May 31 03:20:16 2017\n",
      "PID: 3714                               Python 3.5.3: /opt/conda/bin/python\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/joblib/parallel.py in __call__(self=<joblib.parallel.BatchedCalls object>)\n",
      "    126     def __init__(self, iterator_slice):\n",
      "    127         self.items = list(iterator_slice)\n",
      "    128         self._size = len(self.items)\n",
      "    129 \n",
      "    130     def __call__(self):\n",
      "--> 131         return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "        self.items = [(<function build_IN_levels>, ('chr16:53979176-53980851', [{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1475, 'chr10:100185094-100186268': 1475, 'chr10:100205605-100207134': 1475, 'chr10:100226542-100228626': 1475, 'chr10:100423690-100425020': 1475, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], <networkx.classes.graph.Graph object>, '/input_dir/hg19.fa', '/input_dir/hg19ToHg38.over.chain.gz', '/input_dir/hg38ToHg19.over.chain.gz', '/output_dir//primed/example/', 'example', '/output_dir//primed/example/example_global_graph.json', '/input_dir/TF_RBP_human.ids', 'open_peaks_nodes.bed', 'Adipose_Visceral_Omentum', {'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, '/input_dir/hES_hsa_400kb_xyz.bed', 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed'), {})]\n",
      "    132 \n",
      "    133     def __len__(self):\n",
      "    134         return self._size\n",
      "    135 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/joblib/parallel.py in <listcomp>(.0=<list_iterator object>)\n",
      "    126     def __init__(self, iterator_slice):\n",
      "    127         self.items = list(iterator_slice)\n",
      "    128         self._size = len(self.items)\n",
      "    129 \n",
      "    130     def __call__(self):\n",
      "--> 131         return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "        func = <function build_IN_levels>\n",
      "        args = ('chr16:53979176-53980851', [{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1475, 'chr10:100185094-100186268': 1475, 'chr10:100205605-100207134': 1475, 'chr10:100226542-100228626': 1475, 'chr10:100423690-100425020': 1475, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], <networkx.classes.graph.Graph object>, '/input_dir/hg19.fa', '/input_dir/hg19ToHg38.over.chain.gz', '/input_dir/hg38ToHg19.over.chain.gz', '/output_dir//primed/example/', 'example', '/output_dir//primed/example/example_global_graph.json', '/input_dir/TF_RBP_human.ids', 'open_peaks_nodes.bed', 'Adipose_Visceral_Omentum', {'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, '/input_dir/hES_hsa_400kb_xyz.bed', 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed')\n",
      "        kwargs = {}\n",
      "    132 \n",
      "    133     def __len__(self):\n",
      "    134         return self._size\n",
      "    135 \n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in build_IN_levels(target_node_name='chr16:53979176-53980851', dendogram_com=[{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1528, 'chr10:100185094-100186268': 1528, 'chr10:100205605-100207134': 7231, 'chr10:100226542-100228626': 7231, 'chr10:100423690-100425020': 1528, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], dna_int_graph=<networkx.classes.graph.Graph object>, genome_fa='/input_dir/hg19.fa', chain_file='/input_dir/hg19ToHg38.over.chain.gz', chain_file2='/input_dir/hg38ToHg19.over.chain.gz', out_dir='/output_dir//primed/example/', run_name='example', json_draft_name='/output_dir//primed/example/example_global_graph.json', TF_RBP_ids='/input_dir/TF_RBP_human.ids', open_peaks_file='open_peaks_nodes.bed', picked_tissue='Adipose_Visceral_Omentum', ctcf2bounds={'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, hsa_3D_loc='/input_dir/hES_hsa_400kb_xyz.bed', ucsc_session='http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed')\n",
      "    776         add_snp = True\n",
      "    777     else:\n",
      "    778         add_snp = False\n",
      "    779     source_community_graph = annotate_IN(target_node_name, dendogram_com, level, dna_int_graph, add_open, add_snp,\n",
      "    780                                                   homo_gen, chain_file, chain_file2, TF_RBP_ids, open_peaks_file, picked_tissue, ctcf2bounds, comm_open_out,\n",
      "--> 781                                                   comm_var_out, comm_nodes_out, comm_edges_out, comm_edges_out_bed, ucsc_session_in,out_dir)\n",
      "        comm_var_out = <_io.TextIOWrapper name='/output_dir//primed/exa...y_variants_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        comm_nodes_out = <_io.TextIOWrapper name='/output_dir//primed/exa...nity_nodes_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        comm_edges_out = <_io.TextIOWrapper name='/output_dir//primed/exa...ty_edges_lvl_1.washu' mode='a+' encoding='UTF-8'>\n",
      "        comm_edges_out_bed = <_io.TextIOWrapper name='/output_dir//primed/exa...nity_edges_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        ucsc_session_in = 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...ssionName=primed&position=chr16:54668601-53806737'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    782     #level 1 \n",
      "    783     if len(dendogram_com) > 1:\n",
      "    784         level = 1\n",
      "    785         add_open = False\n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in annotate_IN(target_node_name='chr16:53979176-53980851', dendogram_com=[{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1528, 'chr10:100185094-100186268': 1528, 'chr10:100205605-100207134': 7231, 'chr10:100226542-100228626': 7231, 'chr10:100423690-100425020': 1528, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], level=0, dna_int_graph=<networkx.classes.graph.Graph object>, add_open=True, add_snp=True, homo_gen=<pysam.libcfaidx.FastaFile object>, chain_file='/input_dir/hg19ToHg38.over.chain.gz', chain_file2='/input_dir/hg38ToHg19.over.chain.gz', TF_RBP_ids='/input_dir/TF_RBP_human.ids', open_peaks_file='open_peaks_nodes.bed', picked_tissue='Adipose_Visceral_Omentum', ctcf2bounds={'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, comm_open_out=<_io.TextIOWrapper name='/output_dir//primed/exa...en_regions_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_var_out=<_io.TextIOWrapper name='/output_dir//primed/exa...y_variants_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_nodes_out=<_io.TextIOWrapper name='/output_dir//primed/exa...nity_nodes_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_edges_out=<_io.TextIOWrapper name='/output_dir//primed/exa...ty_edges_lvl_1.washu' mode='a+' encoding='UTF-8'>, comm_edges_out_bed=<_io.TextIOWrapper name='/output_dir//primed/exa...nity_edges_lvl_1.bed' mode='a+' encoding='UTF-8'>, ucsc_session_in='http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...ssionName=primed&position=chr16:54668601-53806737', out_dir='/output_dir//primed/example/')\n",
      "    668         print(\"Annotating Open Regions\")\n",
      "    669         source_community_graph = add_openRegions_predict(source_community_graph, homo_gen, chain_file, chain_file2, TF_RBP_ids, open_peaks_file, target_node_name, level, dict())\n",
      "    670         sys.stdout.flush()\n",
      "    671         if add_snp:\n",
      "    672             print(\"Annotating SNPs/CNVs\")\n",
      "--> 673             source_community_graph = add_variants_predict(source_community_graph, homo_gen, chain_file, TF_RBP_ids, picked_tissue, target_node_name,out_dir)\n",
      "        source_community_graph = <networkx.classes.graph.Graph object>\n",
      "        homo_gen = <pysam.libcfaidx.FastaFile object>\n",
      "        chain_file = '/input_dir/hg19ToHg38.over.chain.gz'\n",
      "        TF_RBP_ids = '/input_dir/TF_RBP_human.ids'\n",
      "        picked_tissue = 'Adipose_Visceral_Omentum'\n",
      "        target_node_name = 'chr16:53979176-53980851'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    674             sys.stdout.flush()\n",
      "    675     \n",
      "    676     #Color Insulated Neighborhood according to node attributes, make mid distance to target node\n",
      "    677 \n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in add_variants_predict(G=<networkx.classes.graph.Graph object>, homo_gen=<pysam.libcfaidx.FastaFile object>, chain_file='/input_dir/hg19ToHg38.over.chain.gz', TF_RBP_ids='/input_dir/TF_RBP_human.ids', tissue_type='Adipose_Visceral_Omentum', target_node_name='chr16:53979176-53980851', out_dir='/output_dir//primed/example/')\n",
      "    297     #predict effect of all variation added to graph using DeepBind and Basset\n",
      "    298     if add_any_SNPs:\n",
      "    299         print(\"Deepbind SNP prediction of: \" + target_node_name)\n",
      "    300         G = deepbind_predict_SNPs(G, homo_gen, TF_RBP_ids,target_node_name)\n",
      "    301         print(\"Basset SNP prediction of: \" + target_node_name)\n",
      "--> 302         G = add_basset_sad_sat(G, homo_gen, target_node_name,out_dir)\n",
      "        G = <networkx.classes.graph.Graph object>\n",
      "        homo_gen = <pysam.libcfaidx.FastaFile object>\n",
      "        target_node_name = 'chr16:53979176-53980851'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    303     try:\n",
      "    304         os.remove(basset_name_out + \"open_variants.vcf\")\n",
      "    305     except OSError:\n",
      "    306         pass\n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in add_basset_sad_sat(G=<networkx.classes.graph.Graph object>, homo_gen=<pysam.libcfaidx.FastaFile object>, target_node_name='chr16:53979176-53980851', out_dir='/output_dir//primed/example/')\n",
      "    401             count += 1\n",
      "    402     cmd = (\"/root/Basset/src/basset_sad.py -f \" + (homo_gen.filename).decode(\"utf-8\") + \" -l 600 -o \" + basset_out_name + \" -t \" + targets_file + \" \" +  model_file + \" \" + basset_out_name + \"/open_variants.vcf\")\n",
      "    403     print(cmd)\n",
      "    404     os.system(cmd)\n",
      "    405     #Pick SNP maximizing for sum of Delta SAD across all cell types in trained model\n",
      "--> 406     sad_table = pd.read_table(basset_out_name + \"/sad_table.txt\", delim_whitespace=True, header = 0)\n",
      "        sad_table = undefined\n",
      "        basset_out_name = '/output_dir//primed/example//chr16_53979176_53980851_basset'\n",
      "    407     sad_table_dense = sad_table.pivot_table(index='rsid', columns='target', values='pred')\n",
      "    408     serial_sad_table = pickle.dumps(sad_table_dense, protocol=0)\n",
      "    409     G.graph[\"sad_table\"] = serial_sad_table\n",
      "    410     sad_table[\"sad_abs\"] = abs(sad_table[\"pred\"])\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in parser_f(filepath_or_buffer='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', sep='\\t', delimiter='\\t', header=0, names=None, index_col=None, usecols=None, squeeze=False, prefix=None, mangle_dupe_cols=True, dtype=None, engine='c', converters=None, true_values=None, false_values=None, skipinitialspace=False, skiprows=None, nrows=None, na_values=None, keep_default_na=True, na_filter=True, verbose=False, skip_blank_lines=True, parse_dates=False, infer_datetime_format=False, keep_date_col=False, date_parser=None, dayfirst=False, iterator=False, chunksize=None, compression='infer', thousands=None, decimal=b'.', lineterminator=None, quotechar='\"', quoting=0, escapechar=None, comment=None, encoding=None, dialect=None, tupleize_cols=False, error_bad_lines=True, warn_bad_lines=True, skipfooter=0, skip_footer=0, doublequote=True, delim_whitespace=True, as_recarray=False, compact_ints=False, use_unsigned=False, low_memory=True, buffer_lines=None, memory_map=False, float_precision=None)\n",
      "    650                     mangle_dupe_cols=mangle_dupe_cols,\n",
      "    651                     tupleize_cols=tupleize_cols,\n",
      "    652                     infer_datetime_format=infer_datetime_format,\n",
      "    653                     skip_blank_lines=skip_blank_lines)\n",
      "    654 \n",
      "--> 655         return _read(filepath_or_buffer, kwds)\n",
      "        filepath_or_buffer = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...}\n",
      "    656 \n",
      "    657     parser_f.__name__ = name\n",
      "    658 \n",
      "    659     return parser_f\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in _read(filepath_or_buffer='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', kwds={'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...})\n",
      "    400     iterator = kwds.get('iterator', False)\n",
      "    401     chunksize = _validate_integer('chunksize', kwds.get('chunksize', None), 1)\n",
      "    402     nrows = _validate_integer('nrows', kwds.get('nrows', None))\n",
      "    403 \n",
      "    404     # Create the parser.\n",
      "--> 405     parser = TextFileReader(filepath_or_buffer, **kwds)\n",
      "        parser = undefined\n",
      "        filepath_or_buffer = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...}\n",
      "    406 \n",
      "    407     if chunksize or iterator:\n",
      "    408         return parser\n",
      "    409 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in __init__(self=<pandas.io.parsers.TextFileReader object>, f='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', engine='c', **kwds={'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...})\n",
      "    757         # might mutate self.engine\n",
      "    758         self.options, self.engine = self._clean_options(options, engine)\n",
      "    759         if 'has_index_names' in kwds:\n",
      "    760             self.options['has_index_names'] = kwds['has_index_names']\n",
      "    761 \n",
      "--> 762         self._make_engine(self.engine)\n",
      "        self._make_engine = <bound method TextFileReader._make_engine of <pandas.io.parsers.TextFileReader object>>\n",
      "        self.engine = 'c'\n",
      "    763 \n",
      "    764     def close(self):\n",
      "    765         self._engine.close()\n",
      "    766 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in _make_engine(self=<pandas.io.parsers.TextFileReader object>, engine='c')\n",
      "    961             self.close()\n",
      "    962             raise\n",
      "    963 \n",
      "    964     def _make_engine(self, engine='c'):\n",
      "    965         if engine == 'c':\n",
      "--> 966             self._engine = CParserWrapper(self.f, **self.options)\n",
      "        self._engine = None\n",
      "        self.f = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        self.options = {'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', 'delim_whitespace': True, ...}\n",
      "    967         else:\n",
      "    968             if engine == 'python':\n",
      "    969                 klass = PythonParser\n",
      "    970             elif engine == 'python-fwf':\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in __init__(self=<pandas.io.parsers.CParserWrapper object>, src='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', **kwds={'allow_leading_cols': True, 'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'decimal': b'.', 'delim_whitespace': True, 'delimiter': '\\t', ...})\n",
      "   1577             kwds['encoding'] = 'utf-8'\n",
      "   1578 \n",
      "   1579         # #2442\n",
      "   1580         kwds['allow_leading_cols'] = self.index_col is not False\n",
      "   1581 \n",
      "-> 1582         self._reader = parsers.TextReader(src, **kwds)\n",
      "        self._reader = undefined\n",
      "        src = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'allow_leading_cols': True, 'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'decimal': b'.', 'delim_whitespace': True, 'delimiter': '\\t', ...}\n",
      "   1583 \n",
      "   1584         # XXX\n",
      "   1585         self.usecols, self.usecols_dtype = _validate_usecols_arg(\n",
      "   1586             self._reader.usecols)\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/_libs/parsers.cpython-35m-x86_64-linux-gnu.so in pandas._libs.parsers.TextReader.__cinit__ (pandas/_libs/parsers.c:4209)()\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/_libs/parsers.cpython-35m-x86_64-linux-gnu.so in pandas._libs.parsers.TextReader._setup_parser_source (pandas/_libs/parsers.c:8873)()\n",
      "\n",
      "FileNotFoundError: File b'/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt' does not exist\n",
      "___________________________________________________________________________\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/root/Euplotid/src/buildAnnotateINs.py\", line 1195, in <module>\n",
      "    TF_RBP_ids, open_peaks_file, picked_tissue, ctcf2bounds, hsa_3D_loc, ucsc_session) for target_node_name in final_in_nodes)\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/joblib/parallel.py\", line 789, in __call__\n",
      "    self.retrieve()\n",
      "  File \"/opt/conda/lib/python3.5/site-packages/joblib/parallel.py\", line 740, in retrieve\n",
      "    raise exception\n",
      "joblib.my_exceptions.JoblibFileNotFoundError: JoblibFileNotFoundError\n",
      "___________________________________________________________________________\n",
      "Multiprocessing exception:\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in <module>()\n",
      "   1190     os.system(\"closest-features --closest --dist \" + open_peaks + \" all_nodes_sorted.bed > \" + open_peaks_file)\n",
      "   1191     \n",
      "   1192     num_cores = multiprocessing.cpu_count()\n",
      "   1193 \n",
      "   1194     results = Parallel(n_jobs=num_cores,temp_folder=out_dir) (delayed(build_IN_levels) (target_node_name, dendogram_com, dna_int_graph, genome_fa, chain_file, chain_file2, out_dir, run_name, json_draft_name,\n",
      "-> 1195                                                                     TF_RBP_ids, open_peaks_file, picked_tissue, ctcf2bounds, hsa_3D_loc, ucsc_session) for target_node_name in final_in_nodes)\n",
      "   1196 \n",
      "   1197     #Add check if succeded?\n",
      "   1198     #Annotate draft genome    \n",
      "   1199     json_comm = open(json_draft_name,\"r\")\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/joblib/parallel.py in __call__(self=Parallel(n_jobs=6), iterable=<generator object <genexpr>>)\n",
      "    784             if pre_dispatch == \"all\" or n_jobs == 1:\n",
      "    785                 # The iterable was consumed all at once by the above for loop.\n",
      "    786                 # No need to wait for async callbacks to trigger to\n",
      "    787                 # consumption.\n",
      "    788                 self._iterating = False\n",
      "--> 789             self.retrieve()\n",
      "        self.retrieve = <bound method Parallel.retrieve of Parallel(n_jobs=6)>\n",
      "    790             # Make sure that we get a last message telling us we are done\n",
      "    791             elapsed_time = time.time() - self._start_time\n",
      "    792             self._print('Done %3i out of %3i | elapsed: %s finished',\n",
      "    793                         (len(self._output), len(self._output),\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "Sub-process traceback:\n",
      "---------------------------------------------------------------------------\n",
      "FileNotFoundError                                  Wed May 31 03:20:16 2017\n",
      "PID: 3714                               Python 3.5.3: /opt/conda/bin/python\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/joblib/parallel.py in __call__(self=<joblib.parallel.BatchedCalls object>)\n",
      "    126     def __init__(self, iterator_slice):\n",
      "    127         self.items = list(iterator_slice)\n",
      "    128         self._size = len(self.items)\n",
      "    129 \n",
      "    130     def __call__(self):\n",
      "--> 131         return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "        self.items = [(<function build_IN_levels>, ('chr16:53979176-53980851', [{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1475, 'chr10:100185094-100186268': 1475, 'chr10:100205605-100207134': 1475, 'chr10:100226542-100228626': 1475, 'chr10:100423690-100425020': 1475, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], <networkx.classes.graph.Graph object>, '/input_dir/hg19.fa', '/input_dir/hg19ToHg38.over.chain.gz', '/input_dir/hg38ToHg19.over.chain.gz', '/output_dir//primed/example/', 'example', '/output_dir//primed/example/example_global_graph.json', '/input_dir/TF_RBP_human.ids', 'open_peaks_nodes.bed', 'Adipose_Visceral_Omentum', {'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, '/input_dir/hES_hsa_400kb_xyz.bed', 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed'), {})]\n",
      "    132 \n",
      "    133     def __len__(self):\n",
      "    134         return self._size\n",
      "    135 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/joblib/parallel.py in <listcomp>(.0=<list_iterator object>)\n",
      "    126     def __init__(self, iterator_slice):\n",
      "    127         self.items = list(iterator_slice)\n",
      "    128         self._size = len(self.items)\n",
      "    129 \n",
      "    130     def __call__(self):\n",
      "--> 131         return [func(*args, **kwargs) for func, args, kwargs in self.items]\n",
      "        func = <function build_IN_levels>\n",
      "        args = ('chr16:53979176-53980851', [{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1475, 'chr10:100185094-100186268': 1475, 'chr10:100205605-100207134': 1475, 'chr10:100226542-100228626': 1475, 'chr10:100423690-100425020': 1475, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], <networkx.classes.graph.Graph object>, '/input_dir/hg19.fa', '/input_dir/hg19ToHg38.over.chain.gz', '/input_dir/hg38ToHg19.over.chain.gz', '/output_dir//primed/example/', 'example', '/output_dir//primed/example/example_global_graph.json', '/input_dir/TF_RBP_human.ids', 'open_peaks_nodes.bed', 'Adipose_Visceral_Omentum', {'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, '/input_dir/hES_hsa_400kb_xyz.bed', 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed')\n",
      "        kwargs = {}\n",
      "    132 \n",
      "    133     def __len__(self):\n",
      "    134         return self._size\n",
      "    135 \n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in build_IN_levels(target_node_name='chr16:53979176-53980851', dendogram_com=[{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1528, 'chr10:100185094-100186268': 1528, 'chr10:100205605-100207134': 7231, 'chr10:100226542-100228626': 7231, 'chr10:100423690-100425020': 1528, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], dna_int_graph=<networkx.classes.graph.Graph object>, genome_fa='/input_dir/hg19.fa', chain_file='/input_dir/hg19ToHg38.over.chain.gz', chain_file2='/input_dir/hg38ToHg19.over.chain.gz', out_dir='/output_dir//primed/example/', run_name='example', json_draft_name='/output_dir//primed/example/example_global_graph.json', TF_RBP_ids='/input_dir/TF_RBP_human.ids', open_peaks_file='open_peaks_nodes.bed', picked_tissue='Adipose_Visceral_Omentum', ctcf2bounds={'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, hsa_3D_loc='/input_dir/hES_hsa_400kb_xyz.bed', ucsc_session='http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...Name=euplotidUCSC&hgS_otherUserSessionName=primed')\n",
      "    776         add_snp = True\n",
      "    777     else:\n",
      "    778         add_snp = False\n",
      "    779     source_community_graph = annotate_IN(target_node_name, dendogram_com, level, dna_int_graph, add_open, add_snp,\n",
      "    780                                                   homo_gen, chain_file, chain_file2, TF_RBP_ids, open_peaks_file, picked_tissue, ctcf2bounds, comm_open_out,\n",
      "--> 781                                                   comm_var_out, comm_nodes_out, comm_edges_out, comm_edges_out_bed, ucsc_session_in,out_dir)\n",
      "        comm_var_out = <_io.TextIOWrapper name='/output_dir//primed/exa...y_variants_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        comm_nodes_out = <_io.TextIOWrapper name='/output_dir//primed/exa...nity_nodes_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        comm_edges_out = <_io.TextIOWrapper name='/output_dir//primed/exa...ty_edges_lvl_1.washu' mode='a+' encoding='UTF-8'>\n",
      "        comm_edges_out_bed = <_io.TextIOWrapper name='/output_dir//primed/exa...nity_edges_lvl_1.bed' mode='a+' encoding='UTF-8'>\n",
      "        ucsc_session_in = 'http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...ssionName=primed&position=chr16:54668601-53806737'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    782     #level 1 \n",
      "    783     if len(dendogram_com) > 1:\n",
      "    784         level = 1\n",
      "    785         add_open = False\n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in annotate_IN(target_node_name='chr16:53979176-53980851', dendogram_com=[{'chr10:100002276-100003364': 108, 'chr10:100021906-100023080': 108, 'chr10:100067964-100070284': 108, 'chr10:100091165-100092468': 108, 'chr10:100174343-100175470': 108, 'chr10:100183011-100184125': 1528, 'chr10:100185094-100186268': 1528, 'chr10:100205605-100207134': 7231, 'chr10:100226542-100228626': 7231, 'chr10:100423690-100425020': 1528, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}, {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 5: 5, 6: 6, 7: 7, 8: 8, 9: 9, ...}], level=0, dna_int_graph=<networkx.classes.graph.Graph object>, add_open=True, add_snp=True, homo_gen=<pysam.libcfaidx.FastaFile object>, chain_file='/input_dir/hg19ToHg38.over.chain.gz', chain_file2='/input_dir/hg38ToHg19.over.chain.gz', TF_RBP_ids='/input_dir/TF_RBP_human.ids', open_peaks_file='open_peaks_nodes.bed', picked_tissue='Adipose_Visceral_Omentum', ctcf2bounds={'chr16:53088027-53090207': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53132274-53134474': ('chr16:53088027-53090207', 'chr16:86527974-86528787'), 'chr16:53193391-53194589': ('chr16:53088027-53090207', 'chr16:53193391-53194589'), 'chr16:53241256-53242869': ('chr16:53055849-53057134', 'chr16:52101824-52104715'), 'chr16:53254174-53255315': ('chr16:53088027-53090207', 'chr16:53254174-53255315'), 'chr16:53776635-53777704': ('chr5:174007939-174009432', 'chr5:130752381-130753830'), 'chr16:53925426-53926800': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:53979176-53980851': ('chr16:54315183-54322626', 'chr16:53925426-53926800'), 'chr16:54315183-54322626': ('chr16:53088027-53090207', 'chr16:86527974-86528787')}, comm_open_out=<_io.TextIOWrapper name='/output_dir//primed/exa...en_regions_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_var_out=<_io.TextIOWrapper name='/output_dir//primed/exa...y_variants_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_nodes_out=<_io.TextIOWrapper name='/output_dir//primed/exa...nity_nodes_lvl_1.bed' mode='a+' encoding='UTF-8'>, comm_edges_out=<_io.TextIOWrapper name='/output_dir//primed/exa...ty_edges_lvl_1.washu' mode='a+' encoding='UTF-8'>, comm_edges_out_bed=<_io.TextIOWrapper name='/output_dir//primed/exa...nity_edges_lvl_1.bed' mode='a+' encoding='UTF-8'>, ucsc_session_in='http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOt...ssionName=primed&position=chr16:54668601-53806737', out_dir='/output_dir//primed/example/')\n",
      "    668         print(\"Annotating Open Regions\")\n",
      "    669         source_community_graph = add_openRegions_predict(source_community_graph, homo_gen, chain_file, chain_file2, TF_RBP_ids, open_peaks_file, target_node_name, level, dict())\n",
      "    670         sys.stdout.flush()\n",
      "    671         if add_snp:\n",
      "    672             print(\"Annotating SNPs/CNVs\")\n",
      "--> 673             source_community_graph = add_variants_predict(source_community_graph, homo_gen, chain_file, TF_RBP_ids, picked_tissue, target_node_name,out_dir)\n",
      "        source_community_graph = <networkx.classes.graph.Graph object>\n",
      "        homo_gen = <pysam.libcfaidx.FastaFile object>\n",
      "        chain_file = '/input_dir/hg19ToHg38.over.chain.gz'\n",
      "        TF_RBP_ids = '/input_dir/TF_RBP_human.ids'\n",
      "        picked_tissue = 'Adipose_Visceral_Omentum'\n",
      "        target_node_name = 'chr16:53979176-53980851'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    674             sys.stdout.flush()\n",
      "    675     \n",
      "    676     #Color Insulated Neighborhood according to node attributes, make mid distance to target node\n",
      "    677 \n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in add_variants_predict(G=<networkx.classes.graph.Graph object>, homo_gen=<pysam.libcfaidx.FastaFile object>, chain_file='/input_dir/hg19ToHg38.over.chain.gz', TF_RBP_ids='/input_dir/TF_RBP_human.ids', tissue_type='Adipose_Visceral_Omentum', target_node_name='chr16:53979176-53980851', out_dir='/output_dir//primed/example/')\n",
      "    297     #predict effect of all variation added to graph using DeepBind and Basset\n",
      "    298     if add_any_SNPs:\n",
      "    299         print(\"Deepbind SNP prediction of: \" + target_node_name)\n",
      "    300         G = deepbind_predict_SNPs(G, homo_gen, TF_RBP_ids,target_node_name)\n",
      "    301         print(\"Basset SNP prediction of: \" + target_node_name)\n",
      "--> 302         G = add_basset_sad_sat(G, homo_gen, target_node_name,out_dir)\n",
      "        G = <networkx.classes.graph.Graph object>\n",
      "        homo_gen = <pysam.libcfaidx.FastaFile object>\n",
      "        target_node_name = 'chr16:53979176-53980851'\n",
      "        out_dir = '/output_dir//primed/example/'\n",
      "    303     try:\n",
      "    304         os.remove(basset_name_out + \"open_variants.vcf\")\n",
      "    305     except OSError:\n",
      "    306         pass\n",
      "\n",
      "...........................................................................\n",
      "/root/Euplotid/src/buildAnnotateINs.py in add_basset_sad_sat(G=<networkx.classes.graph.Graph object>, homo_gen=<pysam.libcfaidx.FastaFile object>, target_node_name='chr16:53979176-53980851', out_dir='/output_dir//primed/example/')\n",
      "    401             count += 1\n",
      "    402     cmd = (\"/root/Basset/src/basset_sad.py -f \" + (homo_gen.filename).decode(\"utf-8\") + \" -l 600 -o \" + basset_out_name + \" -t \" + targets_file + \" \" +  model_file + \" \" + basset_out_name + \"/open_variants.vcf\")\n",
      "    403     print(cmd)\n",
      "    404     os.system(cmd)\n",
      "    405     #Pick SNP maximizing for sum of Delta SAD across all cell types in trained model\n",
      "--> 406     sad_table = pd.read_table(basset_out_name + \"/sad_table.txt\", delim_whitespace=True, header = 0)\n",
      "        sad_table = undefined\n",
      "        basset_out_name = '/output_dir//primed/example//chr16_53979176_53980851_basset'\n",
      "    407     sad_table_dense = sad_table.pivot_table(index='rsid', columns='target', values='pred')\n",
      "    408     serial_sad_table = pickle.dumps(sad_table_dense, protocol=0)\n",
      "    409     G.graph[\"sad_table\"] = serial_sad_table\n",
      "    410     sad_table[\"sad_abs\"] = abs(sad_table[\"pred\"])\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in parser_f(filepath_or_buffer='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', sep='\\t', delimiter='\\t', header=0, names=None, index_col=None, usecols=None, squeeze=False, prefix=None, mangle_dupe_cols=True, dtype=None, engine='c', converters=None, true_values=None, false_values=None, skipinitialspace=False, skiprows=None, nrows=None, na_values=None, keep_default_na=True, na_filter=True, verbose=False, skip_blank_lines=True, parse_dates=False, infer_datetime_format=False, keep_date_col=False, date_parser=None, dayfirst=False, iterator=False, chunksize=None, compression='infer', thousands=None, decimal=b'.', lineterminator=None, quotechar='\"', quoting=0, escapechar=None, comment=None, encoding=None, dialect=None, tupleize_cols=False, error_bad_lines=True, warn_bad_lines=True, skipfooter=0, skip_footer=0, doublequote=True, delim_whitespace=True, as_recarray=False, compact_ints=False, use_unsigned=False, low_memory=True, buffer_lines=None, memory_map=False, float_precision=None)\n",
      "    650                     mangle_dupe_cols=mangle_dupe_cols,\n",
      "    651                     tupleize_cols=tupleize_cols,\n",
      "    652                     infer_datetime_format=infer_datetime_format,\n",
      "    653                     skip_blank_lines=skip_blank_lines)\n",
      "    654 \n",
      "--> 655         return _read(filepath_or_buffer, kwds)\n",
      "        filepath_or_buffer = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...}\n",
      "    656 \n",
      "    657     parser_f.__name__ = name\n",
      "    658 \n",
      "    659     return parser_f\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in _read(filepath_or_buffer='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', kwds={'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...})\n",
      "    400     iterator = kwds.get('iterator', False)\n",
      "    401     chunksize = _validate_integer('chunksize', kwds.get('chunksize', None), 1)\n",
      "    402     nrows = _validate_integer('nrows', kwds.get('nrows', None))\n",
      "    403 \n",
      "    404     # Create the parser.\n",
      "--> 405     parser = TextFileReader(filepath_or_buffer, **kwds)\n",
      "        parser = undefined\n",
      "        filepath_or_buffer = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...}\n",
      "    406 \n",
      "    407     if chunksize or iterator:\n",
      "    408         return parser\n",
      "    409 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in __init__(self=<pandas.io.parsers.TextFileReader object>, f='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', engine='c', **kwds={'as_recarray': False, 'buffer_lines': None, 'chunksize': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': None, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', ...})\n",
      "    757         # might mutate self.engine\n",
      "    758         self.options, self.engine = self._clean_options(options, engine)\n",
      "    759         if 'has_index_names' in kwds:\n",
      "    760             self.options['has_index_names'] = kwds['has_index_names']\n",
      "    761 \n",
      "--> 762         self._make_engine(self.engine)\n",
      "        self._make_engine = <bound method TextFileReader._make_engine of <pandas.io.parsers.TextFileReader object>>\n",
      "        self.engine = 'c'\n",
      "    763 \n",
      "    764     def close(self):\n",
      "    765         self._engine.close()\n",
      "    766 \n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in _make_engine(self=<pandas.io.parsers.TextFileReader object>, engine='c')\n",
      "    961             self.close()\n",
      "    962             raise\n",
      "    963 \n",
      "    964     def _make_engine(self, engine='c'):\n",
      "    965         if engine == 'c':\n",
      "--> 966             self._engine = CParserWrapper(self.f, **self.options)\n",
      "        self._engine = None\n",
      "        self.f = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        self.options = {'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'date_parser': None, 'dayfirst': False, 'decimal': b'.', 'delim_whitespace': True, ...}\n",
      "    967         else:\n",
      "    968             if engine == 'python':\n",
      "    969                 klass = PythonParser\n",
      "    970             elif engine == 'python-fwf':\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/io/parsers.py in __init__(self=<pandas.io.parsers.CParserWrapper object>, src='/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt', **kwds={'allow_leading_cols': True, 'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'decimal': b'.', 'delim_whitespace': True, 'delimiter': '\\t', ...})\n",
      "   1577             kwds['encoding'] = 'utf-8'\n",
      "   1578 \n",
      "   1579         # #2442\n",
      "   1580         kwds['allow_leading_cols'] = self.index_col is not False\n",
      "   1581 \n",
      "-> 1582         self._reader = parsers.TextReader(src, **kwds)\n",
      "        self._reader = undefined\n",
      "        src = '/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt'\n",
      "        kwds = {'allow_leading_cols': True, 'as_recarray': False, 'buffer_lines': None, 'comment': None, 'compact_ints': False, 'compression': None, 'converters': {}, 'decimal': b'.', 'delim_whitespace': True, 'delimiter': '\\t', ...}\n",
      "   1583 \n",
      "   1584         # XXX\n",
      "   1585         self.usecols, self.usecols_dtype = _validate_usecols_arg(\n",
      "   1586             self._reader.usecols)\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/_libs/parsers.cpython-35m-x86_64-linux-gnu.so in pandas._libs.parsers.TextReader.__cinit__ (pandas/_libs/parsers.c:4209)()\n",
      "\n",
      "...........................................................................\n",
      "/opt/conda/lib/python3.5/site-packages/pandas/_libs/parsers.cpython-35m-x86_64-linux-gnu.so in pandas._libs.parsers.TextReader._setup_parser_source (pandas/_libs/parsers.c:8873)()\n",
      "\n",
      "FileNotFoundError: File b'/output_dir//primed/example//chr16_53979176_53980851_basset/sad_table.txt' does not exist\n",
      "___________________________________________________________________________\n"
     ]
    }
   ],
   "source": [
    "!python /root/Euplotid/src/buildAnnotateINs.py \\\n",
    "    -i /input_dir/ -o /output_dir/ \\\n",
    "    -c primed -q Adipose_Visceral_Omentum \\\n",
    "    -n example -t IRX3,FTO,CHD9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/dborgesr/anaconda/lib/python2.7/site-packages/matplotlib/font_manager.py:273: UserWarning:\n",
      "\n",
      "Matplotlib is building the font cache using fc-list. This may take a moment.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Default Insulated Neighborhoods to \n",
    "!python generate_draft_JSONs_all_funcs.py -i /input_data/ -o /root/Euplotid/pipelines/example_output/output_IN_graphs/ \\\n",
    "    -c primed -q Adipose_Visceral_Omentum -n default \n",
    "!python generate_draft_JSONs_all_funcs.py -i /input_data/ -o /root/Euplotid/pipelines/example_output/output_IN_graphs/ \\\n",
    "    -c jurkatt -q Cells_EBV-transformed_lymphocytes -n default \n",
    "!python generate_draft_JSONs_all_funcs.py -i /input_data/ -o /root/Euplotid/pipelines/example_output/output_IN_graphs/ \\\n",
    "    -c neuron -q Brain_Cortex -n default \n",
    "!python generate_draft_JSONs_all_funcs.py -i /input_data/ -o /root/Euplotid/pipelines/example_output/output_IN_graphs/ \\\n",
    "    -c mesc -q Brain_Cortex -n default "
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
