{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build and annotate Insulated Neighborhoods\n",
    "Shows how to build Insulated Neighborhoods using a Louvian Graph based partitioning algorithm and then annotates all Cis-regulatory elements as defined using chromating accessibility using a Neural Network. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Options available to build INs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: buildAnnotatteINs.py [-h] -i INPUT_DIRECTORY -o OUTPUT_DIRECTORY\r\n",
      "                            [-t INSULATED_NEIGH] -c DNA_INT_CELL_TYPE -q\r\n",
      "                            EQTL_TISSUE [-g TOP_GENE_FPKM] -n NAME_OUTPUT\r\n",
      "                            [-m MARKER_REGION]\r\n",
      "\r\n",
      "Pipeline to output graphical genomic model from 2D data\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -i INPUT_DIRECTORY, --input_directory INPUT_DIRECTORY\r\n",
      "                        Where all the input data lives\r\n",
      "  -o OUTPUT_DIRECTORY, --output_directory OUTPUT_DIRECTORY\r\n",
      "                        Output directory of graphs\r\n",
      "  -t INSULATED_NEIGH, --insulated_neigh INSULATED_NEIGH\r\n",
      "                        Comma separated list of Insulated Neighborhoods to\r\n",
      "                        build\r\n",
      "  -c DNA_INT_CELL_TYPE, --dna_int_cell_type DNA_INT_CELL_TYPE\r\n",
      "                        Cell type to use in building Insulated Neighborhoods\r\n",
      "  -q EQTL_TISSUE, --eQTL_tissue EQTL_TISSUE\r\n",
      "                        What tissue from GTeX to do eQTL analysis w/.\r\n",
      "                        Available tissues:\r\n",
      "                        Adipose_Subcutaneous,Adipose_Visceral_Omentum,\r\n",
      "                        Adrenal_Gland, Artery_Aorta, Artery_Coronary,\r\n",
      "                        Artery_Tibial, Brain_Anterior_cingulate_cortex_BA24,\r\n",
      "                        Brain_Caudate_basal_ganglia,\r\n",
      "                        Brain_Cerebellar_Hemisphere, Brain_Cerebellum,\r\n",
      "                        Brain_Cortex, Brain_Frontal_Cortex_BA9,\r\n",
      "                        Brain_Hippocampus, Brain_Hypothalamus,\r\n",
      "                        Brain_Nucleus_accumbens_basal_ganglia,\r\n",
      "                        Brain_Putamen_basal_ganglia,Breast_Mammary_Tissue,\r\n",
      "                        Cells_EBV-transformed_lymphocytes,\r\n",
      "                        Cells_Transformed_fibroblasts, Colon_Sigmoid,\r\n",
      "                        Colon_Transverse, Esophagus_Gastroesophageal_Junction,\r\n",
      "                        Esophagus_Mucosa, Esophagus_Muscularis,\r\n",
      "                        Heart_Atrial_Appendage, Heart_Left_Ventricle, Liver,\r\n",
      "                        Lung, Muscle_Skeletal, Nerve_Tibial, Ovary, Pancreas,\r\n",
      "                        Pituitary, Prostate, Skin_Not_Sun_Exposed_Suprapubic,\r\n",
      "                        Skin_Sun_Exposed_Lower_leg,\r\n",
      "                        Small_Intestine_Terminal_Ileum, Spleen, Stomach,\r\n",
      "                        Testis, Thyroid, Uterus, Vagina, Whole_Blood\r\n",
      "  -g TOP_GENE_FPKM, --top_gene_fpkm TOP_GENE_FPKM\r\n",
      "                        Instead of defining Insulated Neighborhoods take genes\r\n",
      "                        expressed above FPKM cutoff\r\n",
      "  -n NAME_OUTPUT, --name_output NAME_OUTPUT\r\n",
      "                        Output prefix for all output files\r\n",
      "  -m MARKER_REGION, --marker_region MARKER_REGION\r\n",
      "                        Build all Insulated Neighborhoods of genes which fall\r\n",
      "                        within this chromosomal region (ex:\r\n",
      "                        chr1:54996039-55996039)\r\n"
     ]
    }
   ],
   "source": [
    "!python /root/Euplotid/pipelines/buildAnnotatteINs.py --help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example: Build and annotate FTO, IRX3, and CHD9 Insulated Neighborhoods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\r\n",
      "  File \"/root/Euplotid/euplotid/buildAnnotateINs.py\", line 1023, in <module>\r\n",
      "    homo_gen = pysam.FastaFile(genome_fa)\r\n",
      "  File \"pysam/libcfaidx.pyx\", line 114, in pysam.libcfaidx.FastaFile.__cinit__ (pysam/libcfaidx.c:2166)\r\n",
      "  File \"pysam/libcfaidx.pyx\", line 149, in pysam.libcfaidx.FastaFile._open (pysam/libcfaidx.c:2675)\r\n",
      "IOError: file `/annotation_dirhg19.fa` not found\r\n"
     ]
    }
   ],
   "source": [
    "!python /root/Euplotid/euplotid/buildAnnotateINs.py \\\n",
    "    -i /input_dir/ -o /output_dir/ -a /annotation_dir \\\n",
    "    -c primed -q Adipose_Visceral_Omentum \\\n",
    "    -n example -t IRX3,FTO,CHD9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Target Genes: 19\n",
      "Target Genes to build Insulated Neighborhoods: TMEM63A,NANOGNB,DUSP6,NANOG,SOX17,HMGB3,HIF1A-AS2,PRDM14,HIF1A,SMAD3,TBX3,KLF4,CCDC140,PAX3,SOX2,PYCR2,LEFTY1,IRX3,DD413687\n",
      "Checking resolution parameter against CTCF-CTCF loop\n",
      "Annotating 13 Neighborhoods\n",
      "Annotating the following Insulated Neighborhoods: NANOGNB,NANOG|PRDM14|DUSP6|PYCR2,LEFTY1,TMEM63A,LEFTY1|HMGB3|SMAD3,SMAD3|SOX17|TBX3|KLF4|SOX2|HIF1A,HIF1A,HIF1A-AS2|CCDC140,PAX3,DD413687|IRX3\n",
      "Annotating Insulated Neighborhood: NANOGNB,NANOG\n",
      "Annotating Open Regions\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-2...done.\n",
      "Fetching 133 variant(s) . . .\n",
      "Annotating Insulated Neighborhood: PRDM14\n",
      "Annotating Open Regions\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-1...done.\n",
      "Fetching 20 variant(s) . . .\n",
      "Annotating Insulated Neighborhood: PYCR2,LEFTY1,TMEM63A,LEFTY1\n",
      "Annotating Open Regions\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-4...done.\n",
      "No results to return\n",
      "Fetching 54 variant(s) . . .\n",
      "Fetching 80 variant(s) . . .\n",
      "Annotating Insulated Neighborhood: TBX3\n",
      "Annotating Open Regions\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-1...done.\n",
      "Annotating Insulated Neighborhood: SMAD3,SMAD3\n",
      "Annotating Open Regions\n",
      "Fetching 44 variant(s) . . .\n",
      "No results to return\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-2...done.\n",
      "Fetching 27 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Annotating Insulated Neighborhood: CCDC140,PAX3,DD413687\n",
      "Annotating Open Regions\n",
      "Fetching 37 variant(s) . . .\n",
      "Fetching 49 variant(s) . . .\n",
      "Fetching 99 variant(s) . . .\n",
      "Annotating SNPs/CNVs\n",
      "No results to return\n",
      "querying 1-3...done.\n",
      "No results to return\n",
      "Annotating Insulated Neighborhood: HIF1A,HIF1A,HIF1A-AS2\n",
      "Annotating Open Regions\n",
      "Fetching 625 variant(s) . . .\n",
      "Fetching 41 variant(s) . . .\n",
      "Fetching 48 variant(s) . . .\n",
      "No results to return\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-3...done.\n",
      "Fetching 56 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 91 variant(s) . . .\n",
      "Fetching 29 variant(s) . . .\n",
      "Annotating Insulated Neighborhood: DUSP6\n",
      "Annotating Open Regions\n",
      "Annotating SNPs/CNVs\n",
      "No results to return\n",
      "querying 1-1...done.\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr8:70983017-70984817\n",
      "Fetching 37 variant(s) . . .\n",
      "No results to return\n",
      "Annotating Insulated Neighborhood: SOX2\n",
      "Annotating Open Regions\n",
      "No results to return\n",
      "Fetching 24 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr12:7958831-7960358\n",
      "Fetching 38 variant(s) . . .\n",
      "Annotating SNPs/CNVs\n",
      "No results to return\n",
      "querying 1-1...done.\n",
      "Fetching 101 variant(s) . . .\n",
      "Fetching 39 variant(s) . . .\n",
      "No results to return\n",
      "Annotating Insulated Neighborhood: HMGB3\n",
      "Annotating Open Regions\n",
      "Fetching 35 variant(s) . . .\n",
      "Fetching 27 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr14:62216285-62218561\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-1...done.\n",
      "No results to return\n",
      "Fetching 76 variant(s) . . .\n",
      "Annotating Insulated Neighborhood: KLF4\n",
      "Annotating Open Regions\n",
      "No results to return\n",
      "Fetching 7 variant(s) . . .\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-1...done.\n",
      "No results to return\n",
      "Fetching 67 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Annotating Insulated Neighborhood: IRX3\n",
      "Annotating Open Regions\n",
      "Fetching 993 variant(s) . . .\n",
      "Fetching 51 variant(s) . . .\n",
      "Fetching 39 variant(s) . . .\n",
      "Fetching 45 variant(s) . . .\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-1...done.\n",
      "Fetching 20 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 12 variant(s) . . .\n",
      "Annotating Insulated Neighborhood: SOX17\n",
      "Annotating Open Regions\n",
      "Fetching 56 variant(s) . . .\n",
      "No results to return\n",
      "Annotating SNPs/CNVs\n",
      "querying 1-1...done.\n",
      "No results to return\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chrX:150150585-150152257\n",
      "No results to return\n",
      "No results to return\n",
      "Fetching 45 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 50 variant(s) . . .\n",
      "Fetching 87 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Fetching 31 variant(s) . . .\n",
      "Fetching 36 variant(s) . . .\n",
      "Fetching 78 variant(s) . . .\n",
      "Fetching 134 variant(s) . . .\n",
      "Fetching 179 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 30 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "No results to return\n",
      "No results to return\n",
      "Fetching 35 variant(s) . . .\n",
      "Fetching 64 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr1:226068359-226071203\n",
      "Fetching 72 variant(s) . . .\n",
      "Fetching 54 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr8:55370260-55371257\n",
      "No results to return\n",
      "Fetching 47 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr12:89745241-89748630\n",
      "No results to return\n",
      "No results to return\n",
      "No results to return\n",
      "No results to return\n",
      "Fetching 34 variant(s) . . .\n",
      "Fetching 42 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 24 variant(s) . . .\n",
      "Fetching 56 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 59 variant(s) . . .\n",
      "Fetching 46 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr12:115119911-115123081\n",
      "Fetching 42 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 49 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "No results to return\n",
      "Fetching 92 variant(s) . . .\n",
      "Fetching 29 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 78 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 27 variant(s) . . .\n",
      "Fetching 49 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 19 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr15:67355982-67361149\n",
      "No results to return\n",
      "Fetching 26 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 32 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr2:223162665-223165357\n",
      "Fetching 50 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 22 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 73 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Fetching 20 variant(s) . . .\n",
      "Fetching 81 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 23 variant(s) . . .\n",
      "No results to return\n",
      "No results to return\n",
      "Fetching 63 variant(s) . . .\n",
      "Fetching 39 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 141 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 40 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr16:54315183-54322626\n",
      "No results to return\n",
      "Fetching 55 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 50 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 43 variant(s) . . .\n",
      "Basset SNP prediction of: chrX:150150585-150152257\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chrX_150150585_150152257_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chrX_150150585_150152257_basset/open_variants.vcf\n",
      "No results to return\n",
      "No results to return\n",
      "WARNING: Skipping chrX:g.150012231del - neither allele matches reference genome: AT vs TC\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chrX_150150585_150152257_basset/model_in.h5 /output_dir//primed/default//chrX_150150585_150152257_basset/model_out.txt\n",
      "Fetching 54 variant(s) . . .\n",
      "Fetching 27 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr3:181426680-181430164\n",
      "No results to return\n",
      "Fetching 13 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 97 variant(s) . . .\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "No results to return\n",
      "Fetching 66 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 141 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 138 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 50 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 27 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 112 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 71 variant(s) . . .\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "No results to return\n",
      "Fetching 113 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 68 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 36 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 47 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 45 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 50 variant(s) . . .\n",
      "No results to return\n",
      "Fetching 84 variant(s) . . .\n",
      "No results to return\n",
      "Deepbind SNP prediction of: chr9:110248986-110254279\n",
      "Basset SNP prediction of: chr14:62216285-62218561\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr14_62216285_62218561_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr14_62216285_62218561_basset/open_variants.vcf\n",
      "WARNING: Skipping chr14:g.62214687del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr14:g.62214784del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr14:g.62214579del - neither allele matches reference genome: AC vs CC\n",
      "WARNING: Skipping chr14:g.62214472del - neither allele matches reference genome: TA vs AT\n",
      "WARNING: Skipping chr14:g.62214204del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr14:g.62215153_62215155del - neither allele matches reference genome: AAAT vs AATA\n",
      "WARNING: Skipping chr14:g.62215159_62215161del - neither allele matches reference genome: TAAC vs AACA\n",
      "WARNING: Skipping chr14:g.62213865del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr14:g.62214221del - neither allele matches reference genome: AC vs CA\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr14_62216285_62218561_basset/model_in.h5 /output_dir//primed/default//chr14_62216285_62218561_basset/model_out.txt\n",
      "Basset SNP prediction of: chr8:55370260-55371257\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr8_55370260_55371257_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr8_55370260_55371257_basset/open_variants.vcf\n",
      "WARNING: Skipping chr8:g.55374474_55374477del - neither allele matches reference genome: TTAAC vs TAACT\n",
      "WARNING: Skipping chr8:g.55374701del - neither allele matches reference genome: AG vs GG\n",
      "WARNING: Skipping chr8:g.55374545_55374548del - neither allele matches reference genome: CATGG vs ATGGA\n",
      "WARNING: Skipping chr8:g.55374520_55374524del - neither allele matches reference genome: TGTTAA vs GTTAAG\n",
      "WARNING: Skipping chr8:g.55374876_55374877del - neither allele matches reference genome: ACT vs CTC\n",
      "WARNING: Skipping chr8:g.55374830_55374833del - neither allele matches reference genome: CTTAG vs TTAGT\n",
      "WARNING: Skipping chr8:g.55374508_55374509del - neither allele matches reference genome: AAG vs AGA\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr8_55370260_55371257_basset/model_in.h5 /output_dir//primed/default//chr8_55370260_55371257_basset/model_out.txt\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr8:70983017-70984817\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr8_70983017_70984817_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr8_70983017_70984817_basset/open_variants.vcf\n",
      "WARNING: Skipping chr8:g.70668272del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr8:g.70616167_70616169del - neither allele matches reference genome: CATT vs ATTA\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr8_70983017_70984817_basset/model_in.h5 /output_dir//primed/default//chr8_70983017_70984817_basset/model_out.txt\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr12:89745241-89748630\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr12_89745241_89748630_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr12_89745241_89748630_basset/open_variants.vcf\n",
      "WARNING: Skipping chr12:g.89367775del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr12:g.89368130del - neither allele matches reference genome: TA vs AG\n",
      "WARNING: Skipping chr12:g.89368002_89368003del - neither allele matches reference genome: CTT vs TTA\n",
      "WARNING: Skipping chr12:g.89368412_89368416del - neither allele matches reference genome: CTCCTT vs TCCTTT\n",
      "WARNING: Skipping chr12:g.89367775_89367776del - neither allele matches reference genome: CTT vs TTT\n",
      "WARNING: Skipping chr12:g.89367787del - neither allele matches reference genome: TG vs GG\n",
      "WARNING: Skipping chr12:g.89886316_89886317del - neither allele matches reference genome: CAT vs ATA\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr12_89745241_89748630_basset/model_in.h5 /output_dir//primed/default//chr12_89745241_89748630_basset/model_out.txt\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr12:7958831-7960358\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr12_7958831_7960358_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr12_7958831_7960358_basset/open_variants.vcf\n",
      "WARNING: Skipping chr12:g.7933946_7933947del - neither allele matches reference genome: CAA vs AAA\n",
      "WARNING: Skipping chr12:g.7934314_7934315del - neither allele matches reference genome: CAT vs ATG\n",
      "WARNING: Skipping chr12:g.7933809del - neither allele matches reference genome: CT vs TG\n",
      "WARNING: Skipping chr12:g.7933773del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr12:g.7934071del - neither allele matches reference genome: TA vs AA\n",
      "WARNING: Skipping chr12:g.7933945_7933946del - neither allele matches reference genome: TCA vs CAA\n",
      "WARNING: Skipping chr12:g.7934158_7934163del - neither allele matches reference genome: AGAACTT vs GAACTTG\n",
      "WARNING: Skipping chr12:g.7933537del - neither allele matches reference genome: GC vs CC\n",
      "WARNING: Skipping chr12:g.7933933_7933934del - neither allele matches reference genome: CAG vs AGA\n",
      "WARNING: Skipping chr12:g.7943340_7943348del - neither allele matches reference genome: CCCTGCTTTT vs CCTGCTTTTC\n",
      "WARNING: Skipping chr12:g.7943353_7943354del - neither allele matches reference genome: CTT vs TTT\n",
      "WARNING: Skipping chr12:g.7943710del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr12:g.7943706_7943710del - neither allele matches reference genome: CTTTCT vs TTTCTT\n",
      "WARNING: Skipping chr12:g.7943821del - neither allele matches reference genome: CT vs TC\n",
      "WARNING: Skipping chr12:g.7943706_7943711del - neither allele matches reference genome: CTTTCTT vs TTTCTTT\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr12_7958831_7960358_basset/model_in.h5 /output_dir//primed/default//chr12_7958831_7960358_basset/model_out.txt\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr1:226068359-226071203\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr1_226068359_226071203_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr1_226068359_226071203_basset/open_variants.vcf\n",
      "WARNING: Skipping chr1:g.225840609del - neither allele matches reference genome: AG vs GG\n",
      "WARNING: Skipping chr1:g.225840452del - neither allele matches reference genome: AG vs GG\n",
      "WARNING: Skipping chr1:g.225840298del - neither allele matches reference genome: CG vs GG\n",
      "WARNING: Skipping chr1:g.225803730del - neither allele matches reference genome: AC vs CC\n",
      "WARNING: Skipping chr1:g.225850545del - neither allele matches reference genome: TG vs GA\n",
      "WARNING: Skipping chr1:g.225850873_225850875del - neither allele matches reference genome: TAAC vs AACT\n",
      "WARNING: Skipping chr1:g.225850750_225850763del - neither allele matches reference genome: CAAAAAAAAAAAAAA vs AAAAAAAAAAAAAAA\n",
      "WARNING: Skipping chr1:g.225813385_225813388del - neither allele matches reference genome: ATTGT vs TTGTT\n",
      "WARNING: Skipping chr1:g.225813259_225813260del - neither allele matches reference genome: CTT vs TTG\n",
      "WARNING: Skipping chr1:g.225813669_225813671del - neither allele matches reference genome: AAAG vs AAGA\n",
      "WARNING: Skipping chr1:g.226028981_226028984del - neither allele matches reference genome: CTTTA vs TTTAT\n",
      "WARNING: Skipping chr1:g.226028610_226028613del - neither allele matches reference genome: AAGTC vs AGTCA\n",
      "WARNING: Skipping chr1:g.226028796del - neither allele matches reference genome: TG vs GG\n",
      "WARNING: Skipping chr1:g.226028622_226028623del - neither allele matches reference genome: TTC vs TCT\n",
      "WARNING: Skipping chr1:g.226028641_226028643del - neither allele matches reference genome: ATGT vs TGTT\n",
      "WARNING: Skipping chr1:g.226028127del - neither allele matches reference genome: TC vs CC\n",
      "WARNING: Skipping chr1:g.226028749del - neither allele matches reference genome: AT vs TT\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr1_226068359_226071203_basset/model_in.h5 /output_dir//primed/default//chr1_226068359_226071203_basset/model_out.txt\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr15:67355982-67361149\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr15_67355982_67361149_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr15_67355982_67361149_basset/open_variants.vcf\n",
      "WARNING: Skipping chr15:g.67292326del - neither allele matches reference genome: TA vs AT\n",
      "WARNING: Skipping chr15:g.67292329del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr15:g.67292176_67292177del - neither allele matches reference genome: CTG vs TGT\n",
      "WARNING: Skipping chr15:g.67283186_67283189del - neither allele matches reference genome: CAAAG vs AAAGA\n",
      "WARNING: Skipping chr15:g.67283298del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr15:g.67447857del - neither allele matches reference genome: AG vs GG\n",
      "WARNING: Skipping chr15:g.67447903del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr15:g.67447825_67447828del - neither allele matches reference genome: AAGAG vs AGAGA\n",
      "WARNING: Skipping chr15:g.67388045_67388048del - neither allele matches reference genome: ATTTC vs TTTCT\n",
      "WARNING: Skipping chr15:g.67387890_67387891del - neither allele matches reference genome: CTT vs TTT\n",
      "WARNING: Skipping chr15:g.67387890del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr15:g.67286913_67286944del - neither allele matches reference genome: TCAGTGGCTCATGCCTGTAGTCCCAGCACTTTG vs CAGTGGCTCATGCCTGTAGTCCCAGCACTTTGG\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr15_67355982_67361149_basset/model_in.h5 /output_dir//primed/default//chr15_67355982_67361149_basset/model_out.txt\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr16:54315183-54322626\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr16_54315183_54322626_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr16_54315183_54322626_basset/open_variants.vcf\n",
      "WARNING: Skipping chr16:g.53932919_53932920del - neither allele matches reference genome: TAA vs AAA\n",
      "WARNING: Skipping chr16:g.53932930_53932932del - neither allele matches reference genome: CTCT vs TCTT\n",
      "WARNING: Skipping chr16:g.54399504del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr16:g.54261748_54261751del - neither allele matches reference genome: CTATA vs TATAT\n",
      "WARNING: Skipping chr16:g.54261741_54261760del - neither allele matches reference genome: CTATATACTATATATAGTATA vs TATATACTATATATAGTATAT\n",
      "WARNING: Skipping chr16:g.54261644_54261645del - neither allele matches reference genome: CTA vs TAT\n",
      "WARNING: Skipping chr16:g.54261806del - neither allele matches reference genome: CA vs AT\n",
      "WARNING: Skipping chr16:g.54261829_54261841del - neither allele matches reference genome: ATATATGTATACTG vs TATATGTATACTGT\n",
      "WARNING: Skipping chr16:g.53933428del - neither allele matches reference genome: TC vs CC\n",
      "WARNING: Skipping chr16:g.54388922del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr16:g.55188147del - neither allele matches reference genome: CA vs AT\n",
      "WARNING: Skipping chr16:g.55188201del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr16:g.54354078del - neither allele matches reference genome: AC vs CT\n",
      "WARNING: Skipping chr16:g.55124562del - neither allele matches reference genome: AG vs GC\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr16_54315183_54322626_basset/model_in.h5 /output_dir//primed/default//chr16_54315183_54322626_basset/model_out.txt\n",
      "Basset SNP prediction of: chr2:223162665-223165357\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr2_223162665_223165357_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr2_223162665_223165357_basset/open_variants.vcf\n",
      "WARNING: Skipping chr2:g.223522739_223522741del - neither allele matches reference genome: TAAC vs AACA\n",
      "WARNING: Skipping chr2:g.223208242_223208245del - neither allele matches reference genome: TTCTA vs TCTAT\n",
      "WARNING: Skipping chr2:g.222919042_222919047del - neither allele matches reference genome: TCTGTTC vs CTGTTCC\n",
      "WARNING: Skipping chr2:g.222919033del - neither allele matches reference genome: GT vs TT\n",
      "WARNING: Skipping chr2:g.223303104_223303105del - neither allele matches reference genome: ACC vs CCA\n",
      "WARNING: Skipping chr2:g.223302868del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr2:g.223302888del - neither allele matches reference genome: GC vs CC\n",
      "WARNING: Skipping chr2:g.223102049del - neither allele matches reference genome: CA vs AC\n",
      "WARNING: Skipping chr2:g.223102243_223102245del - neither allele matches reference genome: CAAT vs AATC\n",
      "WARNING: Skipping chr2:g.223102046_223102049del - neither allele matches reference genome: GCACA vs CACAC\n",
      "WARNING: Skipping chr2:g.223102046_223102059del - neither allele matches reference genome: GCACACACACACACA vs CACACACACACACAC\n",
      "WARNING: Skipping chr2:g.223102042_223102045del - neither allele matches reference genome: ACATG vs CATGC\n",
      "WARNING: Skipping chr2:g.223102046_223102047del - neither allele matches reference genome: GCA vs CAC\n",
      "WARNING: Skipping chr2:g.223301732del - neither allele matches reference genome: CA vs AG\n",
      "WARNING: Skipping chr2:g.223301183del - neither allele matches reference genome: TA vs AA\n",
      "WARNING: Skipping chr2:g.223301970del - neither allele matches reference genome: AT vs TT\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr2_223162665_223165357_basset/model_in.h5 /output_dir//primed/default//chr2_223162665_223165357_basset/model_out.txt\n",
      "Basset SNP prediction of: chr12:115119911-115123081\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr12_115119911_115123081_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr12_115119911_115123081_basset/open_variants.vcf\n",
      "WARNING: Skipping chr12:g.115514458_115514460del - neither allele matches reference genome: TCAG vs CAGC\n",
      "WARNING: Skipping chr12:g.115514724del - neither allele matches reference genome: CT vs TG\n",
      "WARNING: Skipping chr12:g.115514956_115514958del - neither allele matches reference genome: TAGA vs AGAA\n",
      "WARNING: Skipping chr12:g.115151539del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr12:g.115151895_115151899del - neither allele matches reference genome: TCTTTA vs CTTTAG\n",
      "WARNING: Skipping chr12:g.115151884_115151888del - neither allele matches reference genome: TTCTTA vs TCTTAT\n",
      "WARNING: Skipping chr12:g.115151395del - neither allele matches reference genome: CA vs AA\n",
      "WARNING: Skipping chr12:g.115152053_115152056del - neither allele matches reference genome: TTTTC vs TTTCT\n",
      "WARNING: Skipping chr12:g.115152048del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr12:g.115152023_115152024del - neither allele matches reference genome: CAT vs ATG\n",
      "WARNING: Skipping chr12:g.115559536_115559537del - neither allele matches reference genome: CAA vs AAT\n",
      "WARNING: Skipping chr12:g.115559539_115559543del - neither allele matches reference genome: TCTTTA vs CTTTAC\n",
      "WARNING: Skipping chr12:g.115062282_115062283del - neither allele matches reference genome: CAA vs AAA\n",
      "WARNING: Skipping chr12:g.115062282del - neither allele matches reference genome: CA vs AA\n",
      "WARNING: Skipping chr12:g.115121136_115121139del - neither allele matches reference genome: GAAAA vs AAAAA\n",
      "WARNING: Skipping chr12:g.115121494_115121498del - neither allele matches reference genome: CCCTAT vs CCTATC\n",
      "WARNING: Skipping chr12:g.115121513_115121514del - neither allele matches reference genome: CCT vs CTC\n",
      "WARNING: Skipping chr12:g.115121495_115121498del - neither allele matches reference genome: CCTAT vs CTATC\n",
      "WARNING: Skipping chr12:g.115121128_115121131del - neither allele matches reference genome: AAAAG vs AAAGA\n",
      "WARNING: Skipping chr12:g.115121186del - neither allele matches reference genome: AG vs GG\n",
      "WARNING: Skipping chr12:g.115121496del - neither allele matches reference genome: CT vs TA\n",
      "WARNING: Skipping chr12:g.115121519_115121522del - neither allele matches reference genome: TCTCG vs CTCGC\n",
      "WARNING: Skipping chr12:g.115121132_115121138del - neither allele matches reference genome: GAAAGAAA vs AAAGAAAA\n",
      "WARNING: Skipping chr12:g.115121079_115121082del - neither allele matches reference genome: GTTGT vs TTGTT\n",
      "WARNING: Skipping chr12:g.115543798_115543799del - neither allele matches reference genome: CTT vs TTT\n",
      "WARNING: Skipping chr12:g.115543798del - neither allele matches reference genome: CT vs TT\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr12_115119911_115123081_basset/model_in.h5 /output_dir//primed/default//chr12_115119911_115123081_basset/model_out.txt\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr3:181426680-181430164\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr3_181426680_181430164_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr3_181426680_181430164_basset/open_variants.vcf\n",
      "WARNING: Skipping chr3:g.181648199del - neither allele matches reference genome: AC vs CC\n",
      "WARNING: Skipping chr3:g.181762730del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr3:g.181762664del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr3:g.182229245_182229248del - neither allele matches reference genome: ATTTT vs TTTTA\n",
      "WARNING: Skipping chr3:g.182229660_182229669del - neither allele matches reference genome: GCCTACAGTCC vs CCTACAGTCCA\n",
      "WARNING: Skipping chr3:g.182228434_182228437del - neither allele matches reference genome: CTGTT vs TGTTT\n",
      "WARNING: Skipping chr3:g.182228396del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr3:g.182228510_182228511del - neither allele matches reference genome: GAA vs AAA\n",
      "WARNING: Skipping chr3:g.181673954_181673961del - neither allele matches reference genome: CAAATTGAA vs AAATTGAAA\n",
      "WARNING: Skipping chr3:g.181673960_181673962del - neither allele matches reference genome: GAAA vs AAAG\n",
      "WARNING: Skipping chr3:g.181673954_181673958del - neither allele matches reference genome: CAAATT vs AAATTG\n",
      "WARNING: Skipping chr3:g.181578621_181578628del - neither allele matches reference genome: ACAGGGTTT vs CAGGGTTTC\n",
      "WARNING: Skipping chr3:g.181578711_181578722del - neither allele matches reference genome: GGCATGAGCCACT vs GCATGAGCCACTG\n",
      "WARNING: Skipping chr3:g.181936054_181936056del - neither allele matches reference genome: AGAG vs GAGC\n",
      "WARNING: Skipping chr3:g.181862449del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr3:g.182416175del - neither allele matches reference genome: AC vs CA\n",
      "WARNING: Skipping chr3:g.182415952del - neither allele matches reference genome: TG vs GT\n",
      "WARNING: Skipping chr3:g.181131080del - neither allele matches reference genome: AC vs CG\n",
      "WARNING: Skipping chr3:g.181836534_181836535del - neither allele matches reference genome: AAG vs AGA\n",
      "WARNING: Skipping chr3:g.181836535del - neither allele matches reference genome: AG vs GA\n",
      "WARNING: Skipping chr4:g.145616279_145616282del - neither allele matches reference genome: ACAAG vs CAAGA\n",
      "WARNING: Skipping chr3:g.181575779del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr3:g.181861727_181861728del - neither allele matches reference genome: CAG vs AGA\n",
      "WARNING: Skipping chr3:g.181861661del - neither allele matches reference genome: CT vs TT\n",
      "WARNING: Skipping chr3:g.181861893del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr3:g.181861888del - neither allele matches reference genome: GT vs TG\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr3_181426680_181430164_basset/model_in.h5 /output_dir//primed/default//chr3_181426680_181430164_basset/model_out.txt\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n",
      "Basset SNP prediction of: chr9:110248986-110254279\n",
      "/root/Basset/src/basset_sad.py -f /annotation_dir/hg19.fa -l 600 -o /output_dir//primed/default//chr9_110248986_110254279_basset -t /root/Basset/tutorials/sad_eg/sample_beds.txt /input_dir/pretrained_model.th /output_dir//primed/default//chr9_110248986_110254279_basset/open_variants.vcf\n",
      "WARNING: Skipping chr9:g.110682159del - neither allele matches reference genome: TG vs GC\n",
      "WARNING: Skipping chr9:g.110682137del - neither allele matches reference genome: TA vs AA\n",
      "WARNING: Skipping chr9:g.110682024_110682025del - neither allele matches reference genome: CTA vs TAC\n",
      "WARNING: Skipping chr9:g.110287863del - neither allele matches reference genome: TA vs AA\n",
      "WARNING: Skipping chr9:g.110287791del - neither allele matches reference genome: AG vs GA\n",
      "WARNING: Skipping chr9:g.110288007del - neither allele matches reference genome: CA vs AA\n",
      "WARNING: Skipping chr9:g.110288166_110288168del - neither allele matches reference genome: CAAA vs AAAA\n",
      "WARNING: Skipping chr9:g.110288140_110288143del - neither allele matches reference genome: CTGGG vs TGGGT\n",
      "WARNING: Skipping chr9:g.110288169_110288177del - neither allele matches reference genome: AAACAACAAC vs AACAACAACA\n",
      "WARNING: Skipping chr9:g.110288397_110288409del - neither allele matches reference genome: GTACGGAGATGGAC vs TACGGAGATGGACG\n",
      "WARNING: Skipping chr9:g.110288169_110288183del - neither allele matches reference genome: AAACAACAACAACAAC vs AACAACAACAACAACA\n",
      "WARNING: Skipping chr9:g.110288018del - neither allele matches reference genome: AT vs TA\n",
      "WARNING: Skipping chr9:g.110289066_110289068del - neither allele matches reference genome: ATTT vs TTTT\n",
      "WARNING: Skipping chr9:g.110288900del - neither allele matches reference genome: TA vs AA\n",
      "WARNING: Skipping chr9:g.110940143_110940148del - neither allele matches reference genome: GATAGAT vs ATAGATA\n",
      "WARNING: Skipping chr9:g.111425886_111425887del - neither allele matches reference genome: TGA vs GAG\n",
      "WARNING: Skipping chr9:g.111402663_111402666del - neither allele matches reference genome: CCTCT vs CTCTC\n",
      "WARNING: Skipping chr9:g.110485797del - neither allele matches reference genome: TG vs GG\n",
      "WARNING: Skipping chr9:g.110485907_110485912del - neither allele matches reference genome: CTTTTTT vs TTTTTTT\n",
      "WARNING: Skipping chr9:g.110485902del - neither allele matches reference genome: TC vs CT\n",
      "WARNING: Skipping chr9:g.110485886_110485906del - neither allele matches reference genome: TCTCTCTTTCTTTCTTTCTTTC vs CTCTCTTTCTTTCTTTCTTTCT\n",
      "WARNING: Skipping chr9:g.110485885_110485911del - neither allele matches reference genome: CTCTCTCTTTCTTTCTTTCTTTCTTTTT vs TCTCTCTTTCTTTCTTTCTTTCTTTTTT\n",
      "WARNING: Skipping chr9:g.110485889_110485892del - neither allele matches reference genome: CTCTT vs TCTTT\n",
      "WARNING: Skipping chr9:g.110485902_110485906del - neither allele matches reference genome: TCTTTC vs CTTTCT\n",
      "WARNING: Skipping chr9:g.111218814del - neither allele matches reference genome: TG vs GT\n",
      "WARNING: Skipping chr9:g.111218820_111218821del - neither allele matches reference genome: TAA vs AAA\n",
      "WARNING: Skipping chr9:g.111220122_111220125del - neither allele matches reference genome: AAGAG vs AGAGA\n",
      "WARNING: Skipping chr9:g.111220121_111220122del - neither allele matches reference genome: TAA vs AAG\n",
      "WARNING: Skipping chr9:g.110518021_110518024del - neither allele matches reference genome: GTTTT vs TTTTT\n",
      "WARNING: Skipping chr9:g.110518410_110518413del - neither allele matches reference genome: AGATT vs GATTA\n",
      "WARNING: Skipping chr9:g.110518402_110518405del - neither allele matches reference genome: GGATA vs GATAG\n",
      "WARNING: Skipping chr9:g.110518381_110518384del - neither allele matches reference genome: TACAC vs ACACA\n",
      "WARNING: Skipping chr9:g.111082728_111082735del - neither allele matches reference genome: TAGACAGTC vs AGACAGTCC\n",
      "WARNING: Skipping chr9:g.110452052_110452053del - neither allele matches reference genome: CTT vs TTG\n",
      "WARNING: Skipping chr9:g.110451954_110451955del - neither allele matches reference genome: TAA vs AAA\n",
      "WARNING: Skipping chr9:g.110452358_110452361del - neither allele matches reference genome: GAAAC vs AAACA\n",
      "WARNING: Skipping chr9:g.110400582_110400589del - neither allele matches reference genome: CCTCGCTTT vs CTCGCTTTC\n",
      "WARNING: Skipping chr9:g.110400714_110400716del - neither allele matches reference genome: GCGC vs CGCC\n",
      "WARNING: Skipping chr9:g.110400273_110400274del - neither allele matches reference genome: TCC vs CCG\n",
      "WARNING: Skipping chr9:g.110400713_110400714del - neither allele matches reference genome: TGC vs GCG\n",
      "WARNING: Skipping chr9:g.110400716del - neither allele matches reference genome: GC vs CC\n",
      "WARNING: Skipping chr9:g.110400850_110400852del - neither allele matches reference genome: CAGA vs AGAA\n",
      "WARNING: Skipping chr9:g.110401239_110401240del - neither allele matches reference genome: TTC vs TCA\n",
      "WARNING: Skipping chr9:g.110400665_110400666del - neither allele matches reference genome: CCT vs CTC\n",
      "WARNING: Skipping chr9:g.110401282_110401285del - neither allele matches reference genome: TACAC vs ACACA\n",
      "WARNING: Skipping chr9:g.110401306_110401308del - neither allele matches reference genome: CACA vs ACAC\n",
      "WARNING: Skipping chr9:g.110401282_110401287del - neither allele matches reference genome: TACACAC vs ACACACA\n",
      "WARNING: Skipping chr9:g.110400956_110400957del - neither allele matches reference genome: CTG vs TGG\n",
      "WARNING: Skipping chr9:g.110400628_110400629del - neither allele matches reference genome: TTC vs TCT\n",
      "WARNING: Skipping chr9:g.110400594_110400601del - neither allele matches reference genome: GCTTTCTCT vs CTTTCTCTC\n",
      "WARNING: Skipping chr9:g.110400212_110400214del - neither allele matches reference genome: TTTC vs TTCT\n",
      "WARNING: Skipping chr9:g.110400715del - neither allele matches reference genome: CG vs GC\n",
      "WARNING: Skipping chr9:g.110401282_110401293del - neither allele matches reference genome: TACACACACACAC vs ACACACACACACA\n",
      "WARNING: Skipping chr9:g.110973355_110973362del - neither allele matches reference genome: GGGGTGTGT vs GGGTGTGTG\n",
      "WARNING: Skipping chr9:g.110973355_110973360del - neither allele matches reference genome: GGGGTGT vs GGGTGTG\n",
      "WARNING: Skipping chr9:g.110972813_110972822del - neither allele matches reference genome: AAAAAAAAAAC vs AAAAAAAAACA\n",
      "WARNING: Skipping chr9:g.110973355_110973370del - neither allele matches reference genome: GGGGTGTGTGTGTGTGT vs GGGTGTGTGTGTGTGTG\n",
      "WARNING: Skipping chr9:g.110972274_110972281del - neither allele matches reference genome: CATAAGAGT vs ATAAGAGTA\n",
      "WARNING: Skipping chr9:g.110973384_110973385del - neither allele matches reference genome: GTA vs TAT\n",
      "WARNING: Skipping chr9:g.110973380_110973385del - neither allele matches reference genome: GTGTGTA vs TGTGTAT\n",
      "WARNING: Skipping chr9:g.110973355_110973358del - neither allele matches reference genome: GGGGT vs GGGTG\n",
      "WARNING: Skipping chr9:g.110973370_110973385del - neither allele matches reference genome: GTGTGTGTGTGTGTGTA vs TGTGTGTGTGTGTGTAT\n",
      "WARNING: Skipping chr9:g.110972811_110972822del - neither allele matches reference genome: AAAAAAAAAAAAC vs AAAAAAAAAAACA\n",
      "WARNING: Skipping chr9:g.110973353_110973354del - neither allele matches reference genome: TGG vs GGG\n",
      "WARNING: Skipping chr9:g.110972814_110972822del - neither allele matches reference genome: AAAAAAAAAC vs AAAAAAAACA\n",
      "WARNING: Skipping chr9:g.110973357_110973362del - neither allele matches reference genome: GGTGTGT vs GTGTGTG\n",
      "WARNING: Skipping chr9:g.110973129_110973132del - neither allele matches reference genome: GTCTT vs TCTTT\n",
      "WARNING: Skipping chr9:g.110973378_110973385del - neither allele matches reference genome: GTGTGTGTA vs TGTGTGTAT\n",
      "WARNING: Skipping chr9:g.110973350_110973352del - neither allele matches reference genome: GTTT vs TTTG\n",
      "WARNING: Skipping chr9:g.110272225del - neither allele matches reference genome: CG vs GC\n",
      "WARNING: Skipping chr9:g.110272905del - neither allele matches reference genome: TC vs CA\n",
      "WARNING: Skipping chr9:g.110272497del - neither allele matches reference genome: AG vs GG\n",
      "WARNING: Skipping chr9:g.110450993_110451005del - neither allele matches reference genome: TATGTTCCAGATCC vs ATGTTCCAGATCCA\n",
      "WARNING: Skipping chr9:g.110451058del - neither allele matches reference genome: TA vs AA\n",
      "WARNING: Skipping chr9:g.110451075del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr9:g.110451122del - neither allele matches reference genome: AG vs GG\n",
      "WARNING: Skipping chr9:g.110451086del - neither allele matches reference genome: CA vs AA\n",
      "WARNING: Skipping chr9:g.110769531_110769534del - neither allele matches reference genome: ATTTC vs TTTCT\n",
      "WARNING: Skipping chr9:g.110770377del - neither allele matches reference genome: GC vs CA\n",
      "WARNING: Skipping chr9:g.110770355_110770358del - neither allele matches reference genome: CTTTG vs TTTGT\n",
      "WARNING: Skipping chr9:g.110768815del - neither allele matches reference genome: AT vs TT\n",
      "WARNING: Skipping chr9:g.110769809_110769820del - neither allele matches reference genome: AGAAGTACTAGTT vs GAAGTACTAGTTG\n",
      "WARNING: Skipping chr9:g.110393804del - neither allele matches reference genome: GA vs AA\n",
      "WARNING: Skipping chr9:g.110394134_110394135del - neither allele matches reference genome: CAA vs AAA\n",
      "WARNING: Skipping chr9:g.110673277_110673279del - neither allele matches reference genome: AAAC vs AACA\n",
      "WARNING: Skipping chr9:g.110673481del - neither allele matches reference genome: GA vs AG\n",
      "WARNING: Skipping chr9:g.111482674_111482675del - neither allele matches reference genome: CTG vs TGC\n",
      "WARNING: Skipping chr9:g.110999952_110999954del - neither allele matches reference genome: AAAG vs AAGA\n",
      "WARNING: Skipping chr9:g.111106449_111106451del - neither allele matches reference genome: TAAG vs AAGA\n",
      "WARNING: Skipping chr9:g.111106369_111106371del - neither allele matches reference genome: TGAG vs GAGG\n",
      "/root/Basset/src/basset_predict.lua -rc  /input_dir/pretrained_model.th /output_dir//primed/default//chr9_110248986_110254279_basset/model_in.h5 /output_dir//primed/default//chr9_110248986_110254279_basset/model_out.txt\n",
      "/root/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n",
      "\u001b[0mPredicting sequence 1 variants\u001b[0m\t\n",
      "\u001b[0mPredicting sequence 2 variants\u001b[0m\t\n"
     ]
    }
   ],
   "source": [
    "#Default Insulated Neighborhoods to \n",
    "!python /root/Euplotid/euplotid/buildAnnotateINs.py -i /input_dir/ -o /output_dir/ -a /annotation_dir/ \\\n",
    "    -c primed -q Adipose_Visceral_Omentum -n default \n",
    "#!python generate_draft_JSONs_all_funcs.py -i /input_data/ -o /root/Euplotid/pipelines/example_output/output_IN_graphs/ \\\n",
    "#    -c jurkatt -q Cells_EBV-transformed_lymphocytes -n default \n",
    "#!python generate_draft_JSONs_all_funcs.py -i /input_data/ -o /root/Euplotid/pipelines/example_output/output_IN_graphs/ \\\n",
    "#    -c neuron -q Brain_Cortex -n default \n",
    "#!python generate_draft_JSONs_all_funcs.py -i /input_data/ -o /root/Euplotid/pipelines/example_output/output_IN_graphs/ \\\n",
    "#    -c mesc -q Brain_Cortex -n default "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "jurkatt  local_out  mesc  neuron  primed\r\n"
     ]
    }
   ],
   "source": [
    "!ls /output_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
