{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call DNA-DNA interactions from raw ChiA-PET data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Define data folders, data, and sample name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#remember that fasta should have built bowtie2 indices\n",
    "annotation=\"/annotation_dir/\"\n",
    "tmp=\"/input_data/\"\n",
    "input_dir=\"/output_dir/\"\n",
    "output_dir=\"/output_dir/\"\n",
    "#WT condition\n",
    "input_fq_1=\"rnaseq_mesc_1_1M.fq.gz\"\n",
    "input_fq_2=\"rnaseq_mesc_2_1M.fq.gz\"\n",
    "sample_name=\"rnaseq_mesc_1M_wt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run [ChiA-PET2](https://github.com/GuipengLi/ChIA-PET2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!ChIA-PET2 -g /data/hg19.fa -f /data/fastq/SRR2054934_pass_1.fastq.gz -r /data/fastq/SRR2054934_pass_2.fastq.gz \\\n",
    "    -b /data/hg19.chrom.sizes -A ACGCGATATCTTATCTGACT -B AGTCAGATAAGATATCGCGT \\\n",
    "    -m 1 -s 1 -t 4 -M \"-q .1\" -k 2  -C 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run Origami... kindaaaaa real janky atm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#where all this stuff came from\n",
    "##note: some of this was done sequentially, ie before doing final conversion interaction calling has to finish. \n",
    "##this means that this file cannot be run from start to finish in one go, instead different segments will have to be commented out or in\n",
    "\n",
    "cd /nfs/young_ata4/ENHANCER_RNA/Workshop/CENTRALIZED_FILE_FREEZE/20170412\n",
    "\n",
    "##########################origami##########################\n",
    "#mkdir Origami\n",
    "#cd Origami\n",
    "\n",
    "##lock away the scripts\n",
    "#mkdir Origami_v1.1_freeze\n",
    "#cd Origami_v1.1_freeze\n",
    "#cp -R /home/dsday/projects/ORIGAMI/RunningVersion/origami-v1.1-alpha-2/ origami-v1.1-alpha-2\n",
    "#cp /nfs/young_ata4/ORIGAMI/Data/whitehead-read-name-adjust.sh whitehead-read-name-adjust.sh\n",
    "#cp /nfs/young_ata4/ENHANCER_RNA/Workshop/uniform-calling.sh uniform-calling.sh\n",
    "##cp -R /nfs/young_ata4/ENHANCER_RNA/Workshop/DSD/LoopingAltClassification/ConIntDevel ConInt \n",
    "#copy new version of this\n",
    "#cp -R /nfs/young_ata4/ENHANCER_RNA/Workshop/DSD/LoopingAltClassification/ConIntDevel20170423 20170422_ConInt \n",
    "\n",
    "origami_dir=\"/nfs/young_ata4/ENHANCER_RNA/Workshop/CENTRALIZED_FILE_FREEZE/20170412/Origami/\"\n",
    "origami_vers=\"Origami_v1.1_freeze/origami-v1.1-alpha-2\"\n",
    "##############################################################################################\n",
    "########################                YY1                          #########################\n",
    "##pre-process\n",
    "cd $origami_dir\n",
    "##YY1\n",
    "#mkdir YY1_rep1_rep2_merged\n",
    "cd YY1_rep1_rep2_merged\n",
    "##rep1\n",
    "#ln -s /lab/solexa_public/Young/160122_WIGTC-HISEQ2B_C6E4BANXX/QualityScore/AGGCAG-s_7_1_sequence.txt AGGCAG-s_7_1_sequence.txt\n",
    "#ln -s /lab/solexa_public/Young/160122_WIGTC-HISEQ2B_C6E4BANXX/QualityScore/AGGCAG-s_7_2_sequence.txt AGGCAG-s_7_2_sequence.txt\n",
    "\n",
    "##rep2 \n",
    "#tar --strip-components 5 --to-stdout -xzvf /lab/solexa_public/Young/160308_WIGTC-HISEQA_HKFTJBCXX/QualityScore/AGGCAG-s_1_1_sequence.txt.tar.gz > AGGCAG-s_1_1_sequence.txt\n",
    "#tar --strip-components 5 --to-stdout -xzvf /lab/solexa_public/Young/160308_WIGTC-HISEQA_HKFTJBCXX/QualityScore/AGGCAG-s_1_2_sequence.txt.tar.gz > AGGCAG-s_1_2_sequence.txt\n",
    "#tar --strip-components 5 --to-stdout -xzvf /lab/solexa_public/Young/160308_WIGTC-HISEQA_HKFTJBCXX/QualityScore/AGGCAG-s_2_1_sequence.txt.tar.gz > AGGCAG-s_2_1_sequence.txt\n",
    "#tar --strip-components 5 --to-stdout -xzvf /lab/solexa_public/Young/160308_WIGTC-HISEQA_HKFTJBCXX/QualityScore/AGGCAG-s_2_2_sequence.txt.tar.gz > AGGCAG-s_2_2_sequence.txt\n",
    "\n",
    "##merge rep1 and rep2\n",
    "#cat AGGCAG-s_7_1_sequence.txt AGGCAG-s_1_1_sequence.txt AGGCAG-s_2_1_sequence.txt > YY1.merged_1.txt \n",
    "#cat AGGCAG-s_7_2_sequence.txt AGGCAG-s_1_2_sequence.txt AGGCAG-s_2_2_sequence.txt > YY1.merged_2.txt \n",
    "\n",
    "\n",
    "#call this damn thing\n",
    "#bsub -q14 -e YY1_rep1_rep2_merged.alignment.err -o YY1_rep1_rep2_merged.alignment.out -J YY1_rep1_rep2_alignment \"$origami_dir$origami_vers/bin/origami-alignment --output=$origami_dir/YY1_rep1_rep2_merged --pp=$origami_dir$origami_vers/whitehead-read-name-adjust.sh --macs-gsize=mm /nfs/genomes/mouse_gp_jul_07/bowtie/mm9 YY1.merged_1.txt YY1.merged_2.txt\" \n",
    "##once the first part (trimming + alignment is done) this gets run, dan says the alignment  can also randomly crash out in the middle and need to be restarted from the beginning...\n",
    "#bsub -q14 -e YY1_rep1_rep2_merged.analysis.err -o YY1_rep1_rep2_merged.analysis.out -J YY1_rep1_rep2_analysis -w \"ended(\"YY1_rep1_rep2_alignment\")\" \"$origami_dir$origami_vers/uniform-calling.sh YY1_rep1_rep2_merge\"\n",
    "\n",
    "####do some cleaning up once everything worked\n",
    "#rm -f AGGCAG*\n",
    "#rm -f YY1.merged*\n",
    "#rm -f *fq*c\n",
    "\n",
    "##once analysis is done do some file conversion because CSV files are the dumbest thing ever\n",
    "#cd analysis\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.csv 3 > 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_all.bedpe\n",
    "#$origami_dir$origami_vers/bin/origami-conversion -c washu analysis-results.csv 3 > 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_all.washu\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bed analysis-results.csv 3 > 20170419.YY1_rep1_rep2_merged.origami_v1.1.all.ints_bed\n",
    "##find significant (>0.9) interactions\n",
    "#awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.9) print $0}'  analysis-results.csv  > analysis-results.significant.csv\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.csv 3 > 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.9.bedpe\n",
    "#$origami_dir$origami_vers/bin/origami-conversion -c washu analysis-results.significant.csv 3 > 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.9.washu\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bed analysis-results.significant.csv 3 > 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.9.bed\n",
    "\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.8) print $0}'  analysis-results.csv  > analysis-results.significant.0.8.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.8.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.8.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.7) print $0}'  analysis-results.csv  > analysis-results.significant.0.7.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.7.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.7.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.6) print $0}'  analysis-results.csv  > analysis-results.significant.0.6.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.6.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.6.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.5) print $0}'  analysis-results.csv  > analysis-results.significant.0.5.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.5.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.5.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.4) print $0}'  analysis-results.csv  > analysis-results.significant.0.4.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.4.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.4.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.3) print $0}'  analysis-results.csv  > analysis-results.significant.0.3.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.3.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.3.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.2) print $0}'  analysis-results.csv  > analysis-results.significant.0.2.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.2.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.2.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.1) print $0}'  analysis-results.csv  > analysis-results.significant.0.1.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.1.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.1.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.01) print $0}'  analysis-results.csv  > analysis-results.significant.0.01.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.01.csv 3 >  20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.01.bedpe\n",
    "\n",
    "\n",
    "\n",
    "##############################################################################################\n",
    "########################                CTCF                         #########################\n",
    "cd /nfs/young_ata4/ENHANCER_RNA/Workshop/CENTRALIZED_FILE_FREEZE/20170412/Origami\n",
    "\n",
    "#mkdir CTCF_rep1\n",
    "#cd CTCF_rep1\n",
    "\n",
    "#bsub \"tar --strip-components 5 --to-stdout -xzvf /lab/solexa_public/Young/160523_WIGTC-HISEQ2A_C90H5ANXX/QualityScore/TAAGGC-s_7_1_sequence.txt.tar.gz > CTCF_1.txt\"\n",
    "#bsub \"tar --strip-components 5 --to-stdout -xzvf /lab/solexa_public/Young/160523_WIGTC-HISEQ2A_C90H5ANXX/QualityScore/TAAGGC-s_7_2_sequence.txt.tar.gz > CTCF_2.txt\"\n",
    "#bsub -q14 -e CTCF_rep1_merged.alignment.err -o CTCF_rep1_merged.alignment.out -J CTCF_rep1_alignment \"$origami_dir$origami_vers/bin/origami-alignment --output=$origami_dir/CTCF_rep1 --pp=$origami_dir$origami_vers/whitehead-read-name-adjust.sh --macs-gsize=mm /nfs/genomes/mouse_gp_jul_07/bowtie/mm9 CTCF_1.txt CTCF_2.txt\" \n",
    "##once the first part (trimming + alignment is done) the analysis gets run, dan says the alignment  can also randomly crash out in the middle and need to be restarted from the beginning...\n",
    "#bsub -q14 -e CTCF_rep1.analysis.err -o CTCF_rep1.analysis.out -J CTCF_rep1.analysis -w \"ended(\"CTCF_rep1_alignment\")\" \"$origami_dir$origami_vers/uniform-calling.sh CTCF_rep1\"\n",
    "\n",
    "##do some cleaning up once everything worked\n",
    "#rm -f CTCF_1.txt\n",
    "#rm -f CTCF_2.txt\n",
    "#rm -f *fq*\n",
    "\n",
    "##once analysis is done do some file conversion because CSV files are the dumbest thing ever\n",
    "#cd analysis\n",
    "\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.csv 3 > 20170419.CTCF_rep1.origami_v1.1.ints_all.bedpe\n",
    "#$origami_dir$origami_vers/bin/origami-conversion -c washu analysis-results.csv 3 > 20170419.CTCF_rep1.origami_v1.1.ints_all.washu\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bed analysis-results.csv 3 > 20170419.CTCF_rep1.origami_v1.1.ints_all.bed\n",
    "##find significant (>0.9) interactions\n",
    "#awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.9) print $0}'  analysis-results.csv  > analysis-results.significant.csv\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.9.bedpe\n",
    "#$origami_dir$origami_vers/bin/origami-conversion -c washu analysis-results.significant.csv 3 > 20170419.CTCF_rep1.origami_v1.1.ints_0.9.washu\n",
    "#$origami_dir$origami_vers/bin/origami-conversion bed analysis-results.significant.csv 3 > 20170419.CTCF_rep1.origami_v1.1.ints_0.9.bed\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.8) print $0}'  analysis-results.csv  > analysis-results.significant.0.8.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.8.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.8.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.7) print $0}'  analysis-results.csv  > analysis-results.significant.0.7.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.7.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.7.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.6) print $0}'  analysis-results.csv  > analysis-results.significant.0.6.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.6.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.6.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.5) print $0}'  analysis-results.csv  > analysis-results.significant.0.5.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.5.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.5.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.4) print $0}'  analysis-results.csv  > analysis-results.significant.0.4.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.4.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.4.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.3) print $0}'  analysis-results.csv  > analysis-results.significant.0.3.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.3.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.3.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.2) print $0}'  analysis-results.csv  > analysis-results.significant.0.2.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.2.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.2.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.1) print $0}'  analysis-results.csv  > analysis-results.significant.0.1.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.1.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.1.bedpe\n",
    "\n",
    "awk -F \"\\\"*,\\\"*\" '{if ($9 > 0.01) print $0}'  analysis-results.csv  > analysis-results.significant.0.01.csv\n",
    "$origami_dir$origami_vers/bin/origami-conversion bedpe analysis-results.significant.0.01.csv 3 >  20170419.CTCF_rep1.origami_v1.1.ints_0.01.bedpe\n",
    "\n",
    "\n",
    "#Rscript /nfs/young_ata3/abew/R/bedpetowashu.R  -i 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_all.bedpe  -o 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_all.centered.washu\n",
    "\n",
    "##############################################################################################\n",
    "########################        interaction classification           #########################\n",
    "\n",
    "cd /nfs/young_ata4/ENHANCER_RNA/Workshop/CENTRALIZED_FILE_FREEZE/20170412/Origami\n",
    "#mkdir Int_Class\n",
    "#cd Int_Class\n",
    "#ln -s ../CTCF_rep1/analysis/20170419.CTCF_rep1.origami_v1.1.ints_0.9.bedpe 20170419.CTCF_rep1.origami_v1.1.ints_0.9.bedpe\n",
    "#ln -s ../YY1_rep1_rep2_merged/analysis/20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.9.bedpe 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.9.bedpe\n",
    "#bsub -q14 -e Int_Class.err -o Int_Class.out \"$origami_dir$origami_vers/ConInt/continuous-integration.sh --keep-tmp 20170423.classinfo.xlsx\"\n",
    "\n",
    "mkdir 20170423_DD_Int_Class\n",
    "cd 20170423_DD_Int_Class\n",
    "ln -s ../CTCF_rep1/analysis/20170419.CTCF_rep1.origami_v1.1.ints_0.9.bedpe 20170419.CTCF_rep1.origami_v1.1.ints_0.9.bedpe\n",
    "ln -s ../YY1_rep1_rep2_merged/analysis/20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.9.bedpe 20170419.YY1_rep1_rep2_merged.origami_v1.1.ints_0.9.bedpe\n",
    "bsub -q14 -e Int_Class.err -o Int_Class.out \"$origami_dir$origami_vers/20170422_ConInt/continuous-integration.sh -a -e --keep-tmp 20170423.classinfo.xlsx\"\n",
    "\n",
    "\n",
    "####classify by my script so that we can keep K27Ac that overlap promoters\n",
    "\n",
    "#cd /nfs/young_ata4/ENHANCER_RNA/Workshop/CENTRALIZED_FILE_FREEZE/20170412/Origami\n",
    "#mkdir AW_Int_class\n",
    "#cd AW_Int_class/\n",
    "\n",
    "bash AW_int_class.txt\n",
    "Rscript AW.intclass.R \n",
    "\n",
    "##########################genomic data##########################\n",
    "#cd /nfs/young_ata4/ENHANCER_RNA/Workshop/CENTRALIZED_FILE_FREEZE/20170412\n",
    "#mkdir Genomic_Data\n",
    "#cd Genomic_Data\n",
    "#K27Ac from Xiong\n",
    "#cp /nfs/young_ata4/ENHANCER_RNA/Workshop/Peak_Calls/mES_H3K27ac_wctrl_peaks.bed mES_H3K27ac_wctrl_peaks.bed \n",
    "#OSN from Whyte\n",
    "#cp /nfs/young_ata4/WORKSHOP/SuperEnhancer/mES/mES_HiSeq_OSN_enhs.gff.by_comp mES_HiSeq_OSN_enhs.gff.by_comp\n",
    "#Transcripts from Brian\n",
    "#cp /nfs/genomes/mouse_gp_jul_07/gtf/mm9.refseq.02_01_2017.gtf mm9.refseq.02_01_2017.gtf\n",
    "\n",
    "#Promoters (called by: grep -w transcript /nfs/genomes/mouse_gp_jul_07/gtf/mm9.refseq.02_01_2017.gtf | awk -F'[\\t ]' '{print $1 \"|\" $10 \"|\" $4 \"|\" $5 \"|\" $7}' | sort | uniq | grep -v \"random\" | sed s/\\\"//g | sed s/\\;//g | awk -F'[\\|]' '{\n",
    " # if($5==\"+\")\n",
    "  #  print $1 \"\\t\" $2 \"|\" NR \"\\t\\t\" $3-2000 \"\\t\" $3+2000 \"\\t\\t+\\t\\t\" $2 \"|\" NR\n",
    " # else if($5==\"-\")\n",
    "  #  print $1 \"\\t\" $2 \"|\" NR \"\\t\\t\" $4-2000 \"\\t\" $4+2000 \"\\t\\t-\\t\\t\" $2 \"|\" NR}' > mm9.refseq.02_01_2017.4kbpromsCollapsed mm9.refseq.02_01_2017.4kbproms.gff\n",
    "\n",
    "#CTCF\n",
    "#cp /nfs/young_ata4/ENHANCER_RNA/Workshop/Peak_Calls/mES_CTCF_wctrl_peaks.bed mES_CTCF_wctrl_peaks.bed \n",
    "\n",
    "#insulated neighborhoods - downloaded from website 4/22/17 @ 1 pm\n",
    "#wget http://younglab.wi.mit.edu/INs/mES-data.xlsx\n",
    "#mv mES-data.xlsx Hnisz.review.murine.ESC.IN.xlsx\n",
    "#bullshit code that took me too long to write that removes the bullshit lines that denes put in there and also extracts unique anchors\n",
    "#Rscript reformatIN.R \n",
    "\n",
    "##brian then does some voodoo magic to make a new list b/c things are never simple\n",
    "\n",
    "#intersectBed -u -wa -a /nfs/young_ata4/ENHANCER_RNA/Workshop/Peak_Calls/mES_CTCF_wctrl_peaks.bed -b /nfs/young_ata4/ENHANCER_RNA/Workshop/CENTRALIZED_FILE_FREEZE/20170412/Genomic_Data/Hnisz.review.murine.ESC.IN.uniqueanchors.bed > mES_CTCF_wctrl_peaks.inINanchors.bed\n",
    "#intersectBed -v -b /nfs/young_ata4/ENHANCER_RNA/Workshop/Peak_Calls/mES_H3K27ac_wctrl_peaks.bed -a mES_CTCF_wctrl_peaks.inINanchors.bed >| temp\n",
    "#intersectBed -v -a temp -b mm9.refseq.02_01_2017.4kbproms.gff | awk -F\\\\t '{printf (\"%s\\t%s\\t\\t%8d\\t%8d\\t\\t.\\t\\t%s\\n\", $1,$4, (($3-$2)/2)+($2-2000), (($3-$2)/2)+($2+2000), $4)}' > mES_CTCF_wctrl_peaks.inINanchors.nonEP.4kb.gff\n",
    "#intersectBed -v -a temp -b mm9.refseq.02_01_2017.4kbproms.gff > mES_CTCF_wctrl_peaks.inINanchors.nonEP.gff\n",
    "#rm -f temp.bed\n",
    "\n",
    "#mm9 chrom size\n",
    "#cp /nfs/genomes/mouse_gp_jul_07/anno/chromInfo.txt chromInfo.txt\n",
    "\n",
    "##yy1 \n",
    "cp /nfs/young_ata4/ENHANCER_RNA/Workshop/Peak_Calls/mES_YY1_abChIP_wctrl_1e-9_peaks.bed mES_YY1_abChIP_wctrl_1e-9_peaks.bed"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
