{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Define data folders, data, and sample name\n",
    "#remember that fasta should have built bowtie2 indices\n",
    "annotation=\"/annotation_dir/mm9_norandom\"\n",
    "tmp=\"/input_dir/\"\n",
    "input_dir=\"/output_dir/\"\n",
    "output_dir=\"/output_dir/\"\n",
    "input_fq_1=\"ATAC_mesc_1_5M.fq.gz\"\n",
    "input_fq_2=\"ATAC_mesc_2_5M.fq.gz\"\n",
    "sample_name=\"mesc_atac_5M\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning using bowtie2 w/ paired end\n",
      "5000000 reads; of these:\n",
      "  5000000 (100.00%) were paired; of these:\n",
      "    1827873 (36.56%) aligned concordantly 0 times\n",
      "    1613529 (32.27%) aligned concordantly exactly 1 time\n",
      "    1558598 (31.17%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1827873 pairs aligned concordantly 0 times; of these:\n",
      "      335893 (18.38%) aligned discordantly 1 time\n",
      "    ----\n",
      "    1491980 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      2983960 mates make up the pairs; of these:\n",
      "        2180326 (73.07%) aligned 0 times\n",
      "        129188 (4.33%) aligned exactly 1 time\n",
      "        674446 (22.60%) aligned >1 times\n",
      "78.20% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "#align and sort / filter bam\n",
    "#paired end\n",
    "if [ -e \"$input_dir\"\"$input_fq_2\" ]; then\n",
    "    echo \"Aligning using bowtie2 w/ paired end\"\n",
    "    bowtie2 -p 8 --sensitive -x  $annotation \\\n",
    "        -1 $input_dir$input_fq_1 -2 $input_dir$input_fq_2 | samtools view -bS - > $tmp$sample_name\".bam\"\n",
    "else\n",
    "    echo \"Aligning using bowtie2 w/ single end\"\n",
    "    bowtie2 -p 8 --sensitive -x  $annotation \\\n",
    "        -U $input_dir$input_fq_1 | samtools view -bS - > $tmp$sample_name\".bam\"\n",
    "fi\n",
    "\n",
    "#sort index and filter for canonical chromosomes\n",
    "samtools sort -@ 10 $tmp$sample_name\".bam\" -o $tmp$sample_name\"_sorted.bam\"\n",
    "samtools index \"$tmp$sample_name\"\"_sorted.bam\"\n",
    "samtools view $tmp$sample_name\"_sorted.bam\" -hu chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chrX > \"$output_dir$sample_name\"\"_filt_sorted.bam\"\n",
    "samtools index \"$output_dir$sample_name\"\"_filt_sorted.bam\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR2927023.2702024.1\t99\tchr1\t3015789\t42\t126M\t=\t3015872\t209\tGATGTGGAGAATGTGGAGAAAGAGGAACAATCCACCATTGTTGGTGGGATTACCAGCTTGTACATCCACTCCAGAAATCAGTCTGGTGGTTCCTCAGACTGTACATAGTAGTACTAGAGCATCCTG\tABBBBGGGCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGFGGGF\tAS:i:-15\tXN:i:0\tXM:i:3\tXO:i:0\tXG:i:0\tNM:i:3\tMD:Z:32T31A36G24\tYS:i:-5\tYT:Z:CP\n",
      "SRR2927023.2702024.2\t147\tchr1\t3015872\t42\t126M\t=\t3015789\t-209\tTGGTGGTTCCTCAGACTGTACATAGTAGTACTAGAGCATCCTGCAATACCTCTCCTGTACATATATCTAGATGTTCCAACTGGTAATAAAGACACATGCCCTATTATGTTCATAGCAGCCTTATTT\tGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGDFGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGGGGGGGGCBCCB\tAS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:18G107\tYS:i:-15\tYT:Z:CP\n",
      "SRR2927023.4438235.1\t101\tchr1\t3021827\t0\t*\t=\t3021827\t0\tCTGTGGACCTCACTCAGGATATTATTTTGTAGATCCATTCATTTGCCTGTGAGTTTCATGAAGTCACTATTTTTAATAGCTGAGTAATACTCCATTTTGTAAATATATCAGATTTTCTGTATCTAT\t?BBCBEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFFGGEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGG\tYT:Z:UP\n",
      "SRR2927023.4438235.2\t153\tchr1\t3021827\t42\t126M\t=\t3021827\t0\tCTATTTTTAATAGCTGAGTAATACTCCATTTTGTAAATATATCAGATTTTCTGTATCTATTCCTCTGATGGGAGACATATGGGTTGTTTCTAGCTTTCGGCCATTATAAATATGGCTGATATGAAC\t>GGGGGGGGEGGGGGGGGGGGCGGEEGGGGGGFGGGGFCFB:GGGGGGGFGGGGGGGGFEEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGFGGCCCCC\tAS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:101T24\tYT:Z:UP\n",
      "SRR2927023.489425.2\t163\tchr1\t3025759\t42\t126M\t=\t3025783\t150\tGCTCAACTTTAATATTGGCAACACGCTTTATAAAGAAGAGGGGAAGCCCATTTCTAGCCATGCAAAATTCCTTTGAAGTAGATAGATCAGCCTTCTAGTGGGAACTTGCAAATATCTCAGTTCCAT\tB@ABBGEGG@1CFGG11BFGGGGGGAB/F1FGGC@FGGGGGFBDGBGFFDFGGCGG1FGCF1C1E1CFGFGFGGG@GGGGFGGGEGCEFC>DDGFG>0EGD>FG0FEGFGFDGCGEGGGGGEG=DD\tAS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:112G13\tYS:i:-15\tYT:Z:CP\n",
      "SRR2927023.489425.1\t83\tchr1\t3025783\t42\t126M\t=\t3025759\t-150\tGCTTTATAAAGAAGAGGGGAAGCCCATTTCTAGCCATGCAAAATTCCTTTGAAGTAGATAGATCAGCCTTCTAGTGGGAACTTGCAAATATCTCAGTTCCATGCTAATTTGGGAAAATCTTAGAGG\t0;08FEGGGGGGGFCE0>FGF0EGFGGGGG>BGF=FC1EGGGFGGGFFE>FGF1GGGGGFDF>GDGBGGGGCGGGGF1BGGGGGFGGGGGGEGDDGFE11GGGGGGGGGGGGFGGGGGEGFA0BBA\tAS:i:-15\tXN:i:0\tXM:i:3\tXO:i:0\tXG:i:0\tNM:i:3\tMD:Z:88G18C1C16\tYS:i:-5\tYT:Z:CP\n",
      "SRR2927023.155849.2\t163\tchr1\t3028357\t31\t126M\t=\t3028434\t203\tTCTCTAAACAGAGACAACTATAACAACTAACTCCAGAGATTACCAGATGGAGAAAGGTAAATGTAGGAATTGTACTAACAGGAACCAAGACCACTCACCATCATCAGAACCCAGCACTCCCACTTT\tCCCCCGGGGGGBGFGGGGGG1FGGGGGGGGGGGGGGGGGGGGGGGGGGGFGEGGGGGGGGGGGGGEGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGEGDGDGGGGGGGGBGGGGGG\tAS:i:0\tXS:i:-30\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:126\tYS:i:0\tYT:Z:CP\n",
      "SRR2927023.155849.1\t83\tchr1\t3028434\t31\t126M\t=\t3028357\t-203\tACAGGAACCAAGACCACTCACCATCATCAGAACCCAGCACTCCCACTTTGCCCAGTCCAGGGCACCCCAACACACCTGAAAAGCTAGACCTGGATTTAAAAGCATATCTCATGATGATGGTAGAGG\tGGGGGEGGDFGEGFGGGGGFFGGGGGGGGGGGGCGGGGGG>GEGGFBGGEGGGGGGGFFGGGGGGGGGGEGGF>EGGEGEGGGGGGGGGGGEGGGGGGGGGGGGGGGGGGGGGFGGGEGGDBBBA3\tAS:i:0\tXS:i:-5\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:126\tYS:i:0\tYT:Z:CP\n",
      "SRR2927023.871438.1\t99\tchr1\t3036725\t25\t126M\t=\t3036834\t235\tGTGTATGGGAGTGACTTGGCTATTGTTCTAAGTATTTACTATGTAGACTAGCCCTTAGATTTCTAGCTCTATTCAAGTAAATTTTAATGCCTGATTTCCTTCACTGTCTCTCTTACAATAATAGAG\tABBCBGGGGGGGGGGGGGGGFGGGGGGEEGCFGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGEGGGGGGEFGGFBFGGFFFGGEFGGGGGGGGGGGGEFGGGGGGGC\tAS:i:-10\tXS:i:-37\tXN:i:0\tXM:i:2\tXO:i:0\tXG:i:0\tNM:i:2\tMD:Z:22C80G22\tYS:i:-5\tYT:Z:CP\n",
      "SRR2927023.941947.1\t99\tchr1\t3036765\t21\t126M\t=\t3036821\t182\tATGTAGACTAGCCCTTAGATTTCTAGCTCTATTCAAGTAAATTTTAATGCCTGATTTCCTTCACTGTCTCTCTTACAATAATAGAGGTATGTTTGAATGTTACTGAATAAGTAACATTCTTACTGA\t@BBCCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGDDG@FGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGG\tAS:i:-5\tXS:i:-40\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:63G62\tYS:i:-10\tYT:Z:CP\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n",
      "2994105 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "2942655 + 0 mapped (98.28% : N/A)\n",
      "2994105 + 0 paired in sequencing\n",
      "1496701 + 0 read1\n",
      "1497404 + 0 read2\n",
      "2480490 + 0 properly paired (82.85% : N/A)\n",
      "2891205 + 0 with itself and mate mapped\n",
      "51450 + 0 singletons (1.72% : N/A)\n",
      "110569 + 0 with mate mapped to a different chr\n",
      "15777 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "chr1\t197195432\t1098840\t14022\n",
      "chr2\t181748087\t218434\t5234\n",
      "chr3\t159599783\t109797\t2078\n",
      "chr4\t155630120\t114278\t2194\n",
      "chr5\t152537259\t109942\t2382\n",
      "chr6\t149517037\t115175\t2277\n",
      "chr7\t152524553\t102275\t1965\n",
      "chr8\t131738871\t90764\t1800\n",
      "chr9\t124076172\t118717\t2257\n",
      "chr10\t129993255\t95941\t1777\n",
      "chr11\t121843856\t91955\t1918\n",
      "chr12\t121257530\t96024\t2255\n",
      "chr13\t120284312\t88315\t1624\n",
      "chr14\t125194864\t81129\t1682\n",
      "chr15\t103494974\t72303\t1493\n",
      "chr16\t98319150\t71610\t1284\n",
      "chr17\t95272651\t70427\t1429\n",
      "chr18\t90772031\t65264\t1209\n",
      "chr19\t61342430\t46153\t915\n",
      "chrM\t16299\t0\t0\n",
      "chrX\t166650296\t85312\t1655\n",
      "chrY\t15902555\t0\t0\n",
      "*\t0\t0\t0\n"
     ]
    }
   ],
   "source": [
    "#Take a peek at the bam\n",
    "samtools view \"$output_dir$sample_name\"\"_filt_sorted.bam\"| head\n",
    "#get alignment stats\n",
    "samtools flagstat \"$output_dir$sample_name\"\"_filt_sorted.bam\"\n",
    "samtools idxstats \"$output_dir$sample_name\"\"_filt_sorted.bam\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(py27) (py27) INFO  @ Fri, 26 May 2017 15:40:47: \n",
      "# Command line: callpeak -t /output_dir/mesc_atac_5M_filt_sorted.bam --nomodel --shift -37 -f BAMPE --extsize 73 --broad --keep-dup all -n /input_dir/mesc_atac_5M_MACS\n",
      "# ARGUMENTS LIST:\n",
      "# name = /input_dir/mesc_atac_5M_MACS\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/output_dir/mesc_atac_5M_filt_sorted.bam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Fri, 26 May 2017 15:40:47: #1 read fragment files... \n",
      "INFO  @ Fri, 26 May 2017 15:40:47: #1 read treatment fragments... \n",
      "INFO  @ Fri, 26 May 2017 15:40:59:  1000000 \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #1 mean fragment size is determined as 229 bp from treatment \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #1 fragment size = 229 \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #1  total fragments in treatment: 1240245 \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #1 finished! \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #2 Build Peak Model... \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #2 Skipped... \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #2 Use 229 as fragment length \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #3 Call peaks... \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Fri, 26 May 2017 15:41:03: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Fri, 26 May 2017 15:41:15: #3 Call peaks for each chromosome... \n",
      "INFO  @ Fri, 26 May 2017 15:41:16: #4 Write output xls file... /input_dir/mesc_atac_5M_MACS_peaks.xls \n",
      "INFO  @ Fri, 26 May 2017 15:41:16: #4 Write broad peak in broadPeak format file... /input_dir/mesc_atac_5M_MACS_peaks.broadPeak \n",
      "INFO  @ Fri, 26 May 2017 15:41:16: #4 Write broad peak in bed12/gappedPeak format file... /input_dir/mesc_atac_5M_MACS_peaks.gappedPeak \n",
      "INFO  @ Fri, 26 May 2017 15:41:17: Done! \n",
      "(py27) (py27) Command run:  /root/anaconda3/envs/py27/bin/nucleoatac run --bed /input_dir/mesc_atac_5M_MACS_merged.bed --bam /output_dir/mesc_atac_5M_filt_sorted.bam --fasta /annotation_dir/mm9_norandom.fa --out /input_dir/mesc_atac_5M_nucATAC --cores 8\n",
      "nucleoatac version 0.3.4\n",
      "start run at: 2017-05-26 15:41\n",
      "In file included from /root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1781:0,\n",
      "                 from /root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,\n",
      "                 from /root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,\n",
      "                 from /root/.pyxbld/temp.linux-x86_64-2.7/pyrex/pyatac/fragments.c:435:\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning \"Using deprecated NumPy API, disable it by \" \"#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION\" [-Wcpp]\n",
      " #warning \"Using deprecated NumPy API, disable it by \" \\\n",
      "  ^\n",
      "In file included from /root/.pyxbld/temp.linux-x86_64-2.7/pyrex/pyatac/fragments.c:442:0:\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/htslib_util.h:12:1: warning: function declaration isn't a prototype [-Wstrict-prototypes]\n",
      " int hts_get_verbosity();\n",
      " ^\n",
      "In file included from /root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/pysam_stream.h:4:0,\n",
      "                 from /root/.pyxbld/temp.linux-x86_64-2.7/pyrex/pyatac/fragments.c:452:\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:170:16: warning: 'kseq_init' defined but not used [-Wunused-function]\n",
      "  SCOPE kseq_t *kseq_init(type_t fd)         \\\n",
      "                ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:241:2: note: in expansion of macro '__KSEQ_BASIC'\n",
      "  __KSEQ_BASIC(SCOPE, type_t)     \\\n",
      "  ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:244:35: note: in expansion of macro 'KSEQ_INIT2'\n",
      " #define KSEQ_INIT(type_t, __read) KSEQ_INIT2(static, type_t, __read)\n",
      "                                   ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/pysam_stream.h:9:1: note: in expansion of macro 'KSEQ_INIT'\n",
      " KSEQ_INIT(BGZF *, bgzf_read)\n",
      " ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:176:13: warning: 'kseq_destroy' defined but not used [-Wunused-function]\n",
      "  SCOPE void kseq_destroy(kseq_t *ks)         \\\n",
      "             ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:241:2: note: in expansion of macro '__KSEQ_BASIC'\n",
      "  __KSEQ_BASIC(SCOPE, type_t)     \\\n",
      "  ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:244:35: note: in expansion of macro 'KSEQ_INIT2'\n",
      " #define KSEQ_INIT(type_t, __read) KSEQ_INIT2(static, type_t, __read)\n",
      "                                   ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/pysam_stream.h:9:1: note: in expansion of macro 'KSEQ_INIT'\n",
      " KSEQ_INIT(BGZF *, bgzf_read)\n",
      " ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:190:12: warning: 'kseq_read' defined but not used [-Wunused-function]\n",
      "  SCOPE int kseq_read(kseq_t *seq) \\\n",
      "            ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:242:2: note: in expansion of macro '__KSEQ_READ'\n",
      "  __KSEQ_READ(SCOPE)\n",
      "  ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/include/htslib/htslib/kseq.h:244:35: note: in expansion of macro 'KSEQ_INIT2'\n",
      " #define KSEQ_INIT(type_t, __read) KSEQ_INIT2(static, type_t, __read)\n",
      "                                   ^\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/pysam/pysam_stream.h:9:1: note: in expansion of macro 'KSEQ_INIT'\n",
      " KSEQ_INIT(BGZF *, bgzf_read)\n",
      " ^\n",
      "In file included from /root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1781:0,\n",
      "                 from /root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,\n",
      "                 from /root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,\n",
      "                 from /root/.pyxbld/temp.linux-x86_64-2.7/pyrex/nucleoatac/multinomial_cov.c:435:\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning \"Using deprecated NumPy API, disable it by \" \"#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION\" [-Wcpp]\n",
      " #warning \"Using deprecated NumPy API, disable it by \" \\\n",
      "  ^\n",
      "/root/.pyxbld/temp.linux-x86_64-2.7/pyrex/nucleoatac/multinomial_cov.c: In function '__pyx_pw_10nucleoatac_15multinomial_cov_1calculateCov':\n",
      "/root/.pyxbld/temp.linux-x86_64-2.7/pyrex/nucleoatac/multinomial_cov.c:1704:33: warning: '__pyx_v_value' may be used uninitialized in this function [-Wmaybe-uninitialized]\n",
      "   __pyx_t_2 = PyFloat_FromDouble((__pyx_v_value * __pyx_v_r)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 31, __pyx_L1_error)\n",
      "                                 ^\n",
      "/root/.pyxbld/temp.linux-x86_64-2.7/pyrex/nucleoatac/multinomial_cov.c:1553:50: note: '__pyx_v_value' was declared here\n",
      "   __pyx_t_10nucleoatac_15multinomial_cov_DTYPE_t __pyx_v_value;\n",
      "                                                  ^\n",
      "---------Step1: Computing Occupancy and Nucleosomal Insert Distribution---------\n",
      "Making figure\n",
      "---------Step2: Processing Vplot------------------------------------------------\n",
      "---------Step3: Obtaining nucleosome signal and calling positions---------------\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "---------Step4: Making combined nucleosome position map ------------------------\n",
      "---------Step5: Calling NFR positions-------------------------------------------\n",
      "end run at: 2017-05-26 15:45\n",
      "(py27) "
     ]
    }
   ],
   "source": [
    "#Switch to python2.7 for MACS2 and NucleoATAC\n",
    "source activate py27\n",
    "#Call acessibility peaks, and using those peaks define area around to resolve nucleosomes\n",
    "macs2 callpeak -t \"$output_dir$sample_name\"\"_filt_sorted.bam\" --nomodel --shift -37 -f BAMPE \\\n",
    "    --extsize 73 --broad --keep-dup all -n $tmp\"$sample_name\"\"_MACS\"\n",
    "bedops --range 500:500 --everything $tmp\"$sample_name\"\"_MACS_peaks.broadPeak\" | bedtools merge -i - > $tmp\"$sample_name\"\"_MACS_merged.bed\"\n",
    "nucleoatac run --bed $tmp\"$sample_name\"\"_MACS_merged.bed\" --bam $output_dir\"$sample_name\"\"_filt_sorted.bam\" --fasta \"$annotation\"\".fa\" --out $tmp\"$sample_name\"\"_nucATAC\" --cores 8\n",
    "source deactivate py27"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
