{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Define data folders, data, and sample name\n",
    "#remember that fasta should have built bowtie2 indices\n",
    "annotation=\"/input_data/mm9\"\n",
    "tmp=\"/input_data/\"\n",
    "input_dir=\"/output_graphs/\"\n",
    "output_dir=\"/output_graphs/\"\n",
    "input_fq_1=\"ATAC_mesc_1_5M.fq.gz\"\n",
    "input_fq_2=\"ATAC_mesc_2_5M.fq.gz\"\n",
    "sample_name=\"mesc_atac_5M\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning using bowtie2 w/ paired end\n",
      "5000000 reads; of these:\n",
      "  5000000 (100.00%) were paired; of these:\n",
      "    1824821 (36.50%) aligned concordantly 0 times\n",
      "    1613292 (32.27%) aligned concordantly exactly 1 time\n",
      "    1561887 (31.24%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1824821 pairs aligned concordantly 0 times; of these:\n",
      "      335894 (18.41%) aligned discordantly 1 time\n",
      "    ----\n",
      "    1488927 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      2977854 mates make up the pairs; of these:\n",
      "        2173349 (72.98%) aligned 0 times\n",
      "        128991 (4.33%) aligned exactly 1 time\n",
      "        675514 (22.68%) aligned >1 times\n",
      "78.27% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "#align and sort / filter bam\n",
    "#paired end\n",
    "if [ -e \"$input_dir\"\"$input_fq_2\" ]; then\n",
    "    echo \"Aligning using bowtie2 w/ paired end\"\n",
    "    bowtie2 -p 8 --sensitive -x  $annotation \\\n",
    "        -1 $input_dir$input_fq_1 -2 $input_dir$input_fq_2 | samtools view -bS - > $tmp$sample_name\".bam\"\n",
    "else\n",
    "    echo \"Aligning using bowtie2 w/ single end\"\n",
    "    bowtie2 -p 8 --sensitive -x  $annotation \\\n",
    "        -U $input_dir$input_fq_1 | samtools view -bS - > $tmp$sample_name\".bam\"\n",
    "fi\n",
    "\n",
    "#sort index and filter for canonical chromosomes\n",
    "samtools sort -@ 10 $tmp$sample_name\".bam\" -o $tmp$sample_name\"_sorted.bam\"\n",
    "samtools index \"$tmp$sample_name\"\"_sorted.bam\"\n",
    "samtools view $tmp$sample_name\"_sorted.bam\" -hu chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chrX > \"$output_dir$sample_name\"\"_filt_sorted.bam\"\n",
    "samtools index \"$output_dir$sample_name\"\"_filt_sorted.bam\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR2927023.2702024.1\t99\tchr1\t3015789\t42\t126M\t=\t3015872\t209\tGATGTGGAGAATGTGGAGAAAGAGGAACAATCCACCATTGTTGGTGGGATTACCAGCTTGTACATCCACTCCAGAAATCAGTCTGGTGGTTCCTCAGACTGTACATAGTAGTACTAGAGCATCCTG\tABBBBGGGCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGFGGGF\tAS:i:-15\tXN:i:0\tXM:i:3\tXO:i:0\tXG:i:0\tNM:i:3\tMD:Z:32T31A36G24\tYS:i:-5\tYT:Z:CP\n",
      "SRR2927023.2702024.2\t147\tchr1\t3015872\t42\t126M\t=\t3015789\t-209\tTGGTGGTTCCTCAGACTGTACATAGTAGTACTAGAGCATCCTGCAATACCTCTCCTGTACATATATCTAGATGTTCCAACTGGTAATAAAGACACATGCCCTATTATGTTCATAGCAGCCTTATTT\tGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGDFGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGGGGGGGGCBCCB\tAS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:18G107\tYS:i:-15\tYT:Z:CP\n",
      "SRR2927023.4438235.1\t101\tchr1\t3021827\t0\t*\t=\t3021827\t0\tCTGTGGACCTCACTCAGGATATTATTTTGTAGATCCATTCATTTGCCTGTGAGTTTCATGAAGTCACTATTTTTAATAGCTGAGTAATACTCCATTTTGTAAATATATCAGATTTTCTGTATCTAT\t?BBCBEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFFGGEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGG\tYT:Z:UP\n",
      "SRR2927023.4438235.2\t153\tchr1\t3021827\t42\t126M\t=\t3021827\t0\tCTATTTTTAATAGCTGAGTAATACTCCATTTTGTAAATATATCAGATTTTCTGTATCTATTCCTCTGATGGGAGACATATGGGTTGTTTCTAGCTTTCGGCCATTATAAATATGGCTGATATGAAC\t>GGGGGGGGEGGGGGGGGGGGCGGEEGGGGGGFGGGGFCFB:GGGGGGGFGGGGGGGGFEEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGFGGCCCCC\tAS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:101T24\tYT:Z:UP\n",
      "SRR2927023.489425.2\t163\tchr1\t3025759\t42\t126M\t=\t3025783\t150\tGCTCAACTTTAATATTGGCAACACGCTTTATAAAGAAGAGGGGAAGCCCATTTCTAGCCATGCAAAATTCCTTTGAAGTAGATAGATCAGCCTTCTAGTGGGAACTTGCAAATATCTCAGTTCCAT\tB@ABBGEGG@1CFGG11BFGGGGGGAB/F1FGGC@FGGGGGFBDGBGFFDFGGCGG1FGCF1C1E1CFGFGFGGG@GGGGFGGGEGCEFC>DDGFG>0EGD>FG0FEGFGFDGCGEGGGGGEG=DD\tAS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:112G13\tYS:i:-15\tYT:Z:CP\n",
      "SRR2927023.489425.1\t83\tchr1\t3025783\t42\t126M\t=\t3025759\t-150\tGCTTTATAAAGAAGAGGGGAAGCCCATTTCTAGCCATGCAAAATTCCTTTGAAGTAGATAGATCAGCCTTCTAGTGGGAACTTGCAAATATCTCAGTTCCATGCTAATTTGGGAAAATCTTAGAGG\t0;08FEGGGGGGGFCE0>FGF0EGFGGGGG>BGF=FC1EGGGFGGGFFE>FGF1GGGGGFDF>GDGBGGGGCGGGGF1BGGGGGFGGGGGGEGDDGFE11GGGGGGGGGGGGFGGGGGEGFA0BBA\tAS:i:-15\tXN:i:0\tXM:i:3\tXO:i:0\tXG:i:0\tNM:i:3\tMD:Z:88G18C1C16\tYS:i:-5\tYT:Z:CP\n",
      "SRR2927023.155849.2\t163\tchr1\t3028357\t31\t126M\t=\t3028434\t203\tTCTCTAAACAGAGACAACTATAACAACTAACTCCAGAGATTACCAGATGGAGAAAGGTAAATGTAGGAATTGTACTAACAGGAACCAAGACCACTCACCATCATCAGAACCCAGCACTCCCACTTT\tCCCCCGGGGGGBGFGGGGGG1FGGGGGGGGGGGGGGGGGGGGGGGGGGGFGEGGGGGGGGGGGGGEGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGEGDGDGGGGGGGGBGGGGGG\tAS:i:0\tXS:i:-30\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:126\tYS:i:0\tYT:Z:CP\n",
      "SRR2927023.155849.1\t83\tchr1\t3028434\t31\t126M\t=\t3028357\t-203\tACAGGAACCAAGACCACTCACCATCATCAGAACCCAGCACTCCCACTTTGCCCAGTCCAGGGCACCCCAACACACCTGAAAAGCTAGACCTGGATTTAAAAGCATATCTCATGATGATGGTAGAGG\tGGGGGEGGDFGEGFGGGGGFFGGGGGGGGGGGGCGGGGGG>GEGGFBGGEGGGGGGGFFGGGGGGGGGGEGGF>EGGEGEGGGGGGGGGGGEGGGGGGGGGGGGGGGGGGGGGFGGGEGGDBBBA3\tAS:i:0\tXS:i:-5\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:126\tYS:i:0\tYT:Z:CP\n",
      "SRR2927023.2881758.1\t113\tchr1\t3032143\t6\t50M1D76M\tchrY_random\t52930665\t0\tGAAGGAGCAATTTGCAAATTCAGCTGGAATAACAAAAAACCTAGGATAGCAAATCTCTTCTCAAGGATAAAAGAACCTCTGGTGGAATCACCATGCCTGACCTAAAGCTTTACTACAGGGCAATTG\t0CD@F8800C0F@8<00;00>F0FGE0000GGGE0F@111CGEGG@1E>11:<1=11<0C1EF1D>:11?111>10>0@E@DDEECGGGGGGGG@B:D>E>BD?11@=;C>GGF@01CEGGBABA3\tAS:i:-20\tXN:i:0\tXM:i:3\tXO:i:1\tXG:i:1\tNM:i:4\tMD:Z:50^A3A0G63A7\tYT:Z:UP\n",
      "SRR2927023.871438.1\t99\tchr1\t3036725\t25\t126M\t=\t3036834\t235\tGTGTATGGGAGTGACTTGGCTATTGTTCTAAGTATTTACTATGTAGACTAGCCCTTAGATTTCTAGCTCTATTCAAGTAAATTTTAATGCCTGATTTCCTTCACTGTCTCTCTTACAATAATAGAG\tABBCBGGGGGGGGGGGGGGGFGGGGGGEEGCFGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGEGGGGGGEFGGFBFGGFFFGGEFGGGGGGGGGGGGEFGGGGGGGC\tAS:i:-10\tXS:i:-37\tXN:i:0\tXM:i:2\tXO:i:0\tXG:i:0\tNM:i:2\tMD:Z:22C80G22\tYS:i:-5\tYT:Z:CP\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n",
      "2983038 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "2932338 + 0 mapped (98.30% : N/A)\n",
      "2983038 + 0 paired in sequencing\n",
      "1491174 + 0 read1\n",
      "1491864 + 0 read2\n",
      "2471794 + 0 properly paired (82.86% : N/A)\n",
      "2881638 + 0 with itself and mate mapped\n",
      "50700 + 0 singletons (1.70% : N/A)\n",
      "110670 + 0 with mate mapped to a different chr\n",
      "15835 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "chr1\t197195432\t1097299\t13969\n",
      "chr2\t181748087\t219193\t5223\n",
      "chr3\t159599783\t109510\t2042\n",
      "chr4\t155630120\t113815\t2167\n",
      "chr5\t152537259\t108676\t2324\n",
      "chr6\t149517037\t114944\t2263\n",
      "chr7\t152524553\t102114\t1946\n",
      "chr8\t131738871\t89735\t1757\n",
      "chr9\t124076172\t116391\t2234\n",
      "chr10\t129993255\t95976\t1766\n",
      "chr11\t121843856\t91893\t1914\n",
      "chr12\t121257530\t96650\t2203\n",
      "chr13\t120284312\t88108\t1607\n",
      "chr14\t125194864\t80842\t1658\n",
      "chr15\t103494974\t72155\t1484\n",
      "chr16\t98319150\t71380\t1258\n",
      "chr17\t95272651\t70112\t1394\n",
      "chr18\t90772031\t65068\t1200\n",
      "chr19\t61342430\t46101\t904\n",
      "chrX\t166650296\t82376\t1387\n",
      "chrY\t15902555\t0\t0\n",
      "chrM\t16299\t0\t0\n",
      "chr13_random\t400311\t0\t0\n",
      "chr16_random\t3994\t0\t0\n",
      "chr17_random\t628739\t0\t0\n",
      "chr1_random\t1231697\t0\t0\n",
      "chr3_random\t41899\t0\t0\n",
      "chr4_random\t160594\t0\t0\n",
      "chr5_random\t357350\t0\t0\n",
      "chr7_random\t362490\t0\t0\n",
      "chr8_random\t849593\t0\t0\n",
      "chr9_random\t449403\t0\t0\n",
      "chrUn_random\t5900358\t0\t0\n",
      "chrX_random\t1785075\t0\t0\n",
      "chrY_random\t58682461\t0\t0\n",
      "*\t0\t0\t0\n"
     ]
    }
   ],
   "source": [
    "#Take a peek at the bam\n",
    "samtools view \"$output_dir$sample_name\"\"_filt_sorted.bam\"| head\n",
    "#get alignment stats\n",
    "samtools flagstat \"$output_dir$sample_name\"\"_filt_sorted.bam\"\n",
    "samtools idxstats \"$output_dir$sample_name\"\"_filt_sorted.bam\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(py27) (py27) INFO  @ Thu, 25 May 2017 06:42:02: \n",
      "# Command line: callpeak -t /output_graphs/mesc_atac_5M_filt_sorted.bam --nomodel --shift -37 -f BAMPE --extsize 73 --broad --keep-dup all -n /input_data/mesc_atac_5M_MACS\n",
      "# ARGUMENTS LIST:\n",
      "# name = /input_data/mesc_atac_5M_MACS\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/output_graphs/mesc_atac_5M_filt_sorted.bam']\n",
      "# control file = None\n",
      "# effective genome size = 2.70e+09\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      " \n",
      "INFO  @ Thu, 25 May 2017 06:42:02: #1 read fragment files... \n",
      "INFO  @ Thu, 25 May 2017 06:42:02: #1 read treatment fragments... \n",
      "INFO  @ Thu, 25 May 2017 06:42:17:  1000000 \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #1 mean fragment size is determined as 233 bp from treatment \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #1 fragment size = 233 \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #1  total fragments in treatment: 1235897 \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #1 finished! \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #2 Build Peak Model... \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #2 Skipped... \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #2 Use 233 as fragment length \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #3 Call peaks... \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Thu, 25 May 2017 06:42:21: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Thu, 25 May 2017 06:42:31: #3 Call peaks for each chromosome... \n",
      "INFO  @ Thu, 25 May 2017 06:42:32: #4 Write output xls file... /input_data/mesc_atac_5M_MACS_peaks.xls \n",
      "INFO  @ Thu, 25 May 2017 06:42:32: #4 Write broad peak in broadPeak format file... /input_data/mesc_atac_5M_MACS_peaks.broadPeak \n",
      "INFO  @ Thu, 25 May 2017 06:42:32: #4 Write broad peak in bed12/gappedPeak format file... /input_data/mesc_atac_5M_MACS_peaks.gappedPeak \n",
      "INFO  @ Thu, 25 May 2017 06:42:32: Done! \n",
      "(py27) (py27) Command run:  /root/anaconda3/envs/py27/bin/nucleoatac run --bed /input_data/mesc_atac_5M_MACS_merged.bed --bam /output_graphs/mesc_atac_5M_filt_sorted.bam --fasta /input_data/mm9.fa --out /input_data/mesc_atac_5M_nucATAC --cores 8\n",
      "nucleoatac version 0.3.4\n",
      "start run at: 2017-05-25 06:42\n",
      "---------Step1: Computing Occupancy and Nucleosomal Insert Distribution---------\n",
      "Making figure\n",
      "---------Step2: Processing Vplot------------------------------------------------\n",
      "---------Step3: Obtaining nucleosome signal and calling positions---------------\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: divide by zero encountered in log\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:120: RuntimeWarning: invalid value encountered in multiply\n",
      "  nuc_lik = np.sum(np.log(nuc_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: divide by zero encountered in log\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "/root/anaconda3/envs/py27/lib/python2.7/site-packages/nucleoatac/NucleosomeCalling.py:121: RuntimeWarning: invalid value encountered in multiply\n",
      "  null_lik = np.sum(np.log(null_model) * mat)\n",
      "---------Step4: Making combined nucleosome position map ------------------------\n",
      "---------Step5: Calling NFR positions-------------------------------------------\n",
      "end run at: 2017-05-25 06:45\n",
      "(py27) "
     ]
    }
   ],
   "source": [
    "#Switch to python2.7 for MACS2 and NucleoATAC\n",
    "source activate py27\n",
    "#Call acessibility peaks, and using those peaks define area around to resolve nucleosomes\n",
    "macs2 callpeak -t \"$output_dir$sample_name\"\"_filt_sorted.bam\" --nomodel --shift -37 -f BAMPE \\\n",
    "    --extsize 73 --broad --keep-dup all -n $tmp\"$sample_name\"\"_MACS\"\n",
    "bedops --range 500:500 --everything $tmp\"$sample_name\"\"_MACS_peaks.broadPeak\" | bedtools merge -i - > $tmp\"$sample_name\"\"_MACS_merged.bed\"\n",
    "nucleoatac run --bed $tmp\"$sample_name\"\"_MACS_merged.bed\" --bam $output_dir\"$sample_name\"\"_filt_sorted.bam\" --fasta \"$annotation\"\".fa\" --out $tmp\"$sample_name\"\"_nucATAC\" --cores 8\n",
    "source deactivate py27"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
